Index: alsa-utils-1.0.18/alsactl/state.c
===================================================================
--- alsa-utils-1.0.18.orig/alsactl/state.c	2009-01-12 18:55:02.000000000 -0500
+++ alsa-utils-1.0.18/alsactl/state.c	2009-01-12 18:55:33.000000000 -0500
@@ -1166,7 +1166,7 @@
 		return -EINVAL;
 	}
 	numid = atoi(id);
-	if (numid > *maxnumid)
+	if ((int)numid > *maxnumid)
 		*maxnumid = numid;
 	snd_config_for_each(i, next, control) {
 		snd_config_t *n = snd_config_iterator_entry(i);
@@ -1409,7 +1409,7 @@
 	snd_ctl_card_info_t *info;
 	snd_config_t *control;
 	snd_config_iterator_t i, next;
-	int err, maxnumid;
+	int err, maxnumid = -1;
 	char name[32], tmpid[16];
 	const char *id;
 	snd_ctl_card_info_alloca(&info);
@@ -1453,7 +1453,7 @@
 
 	/* check if we have additional controls in driver */
 	/* in this case we should go through init procedure */
-	if (!doit) {
+	if (!doit && maxnumid >= 0) {
 		snd_ctl_elem_id_t *id;
 		snd_ctl_elem_info_t *info;
 		snd_ctl_elem_id_alloca(&id);
