diff -ruNad alsa-driver-1.0.15.orig/alsa-kernel/pci/atiixp.c alsa-driver-1.0.15/alsa-kernel/pci/atiixp.c
--- alsa-driver-1.0.15.orig/alsa-kernel/pci/atiixp.c	2007-12-31 22:01:41.000000000 +0800
+++ alsa-driver-1.0.15/alsa-kernel/pci/atiixp.c	2007-12-31 22:02:23.000000000 +0800
@@ -291,6 +291,7 @@
 	{ 0x1002, 0x4341, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, /* SB200 */
 	{ 0x1002, 0x4361, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, /* SB300 */
 	{ 0x1002, 0x4370, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, /* SB400 */
+	{ 0x1002, 0x4382, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, /* SB600 */
 	{ 0, }
 };
 
