Index: alsa-driver-1.0.18.dfsg/configure
===================================================================
--- alsa-driver-1.0.18.dfsg.orig/configure	2008-11-02 18:27:24.000000000 +0100
+++ alsa-driver-1.0.18.dfsg/configure	2008-11-02 18:27:24.000000000 +0100
@@ -1671,16 +1671,16 @@
                             pc98-cs4232, msnd-pinnacle, ad1889, als300,
                             als4000, ali5451, atiixp, atiixp-modem, au8810,
                             au8820, au8830, aw2, azt3328, bt87x, ca0106,
-                            cmipci, oxygen, cs4281, cs46xx, cs5530,
+                            cmipci, oxygen, cs4281,
                             cs5535audio, darla20, gina20, layla20, darla24,
                             gina24, layla24, mona, mia, echo3g, indigo,
                             indigoio, indigodj, emu10k1, emu10k1x, ens1370,
                             ens1371, es1938, es1968, fm801, hda-intel, hdsp,
                             hdspm, hifier, ice1712, ice1724, intel8x0,
-                            intel8x0m, korg1212, maestro3, mixart, nm256,
+                            intel8x0m, mixart, nm256,
                             pcxhr, riptide, rme32, rme96, rme9652, sis7019,
                             sonicvibes, trident, via82xx, via82xx-modem,
-                            virtuoso, vx222, ymfpci, pdplus, asihpi,
+                            virtuoso, vx222, pdplus, asihpi,
                             powermac, ps3, aoa, aoa-fabric-layout, aoa-onyx,
                             aoa-tas, aoa-toonie, aoa-soundbus,
                             aoa-soundbus-i2s, sa11xx-uda1341, armaaci,
@@ -1709,7 +1709,7 @@
                             verbose-procfs, verbose-printk, pcm-xrun-debug,
                             ac97-power-save, sb16-csp,
                             wavefront-firmware-in-kernel, bt87x-overclock,
-                            cs46xx-new-dsp, fm801-tea575x-bool, hda-hwdep,
+                            fm801-tea575x-bool, hda-hwdep,
                             hda-input-beep, hda-codec-realtek,
                             hda-codec-analog, hda-codec-sigmatel,
                             hda-codec-via, hda-codec-atihdmi,
@@ -14624,25 +14624,6 @@
       CONFIG_SND_CS4281="m"
     fi
   fi
-  if alsa_check_kconfig_card "cs46xx"; then
-    if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
-      ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ); then
-      CONFIG_SND_TIMER="m"
-      CONFIG_SND_PCM="m"
-      test "$kversion.$kpatchlevel" = "2.6" -a $ksublevel -ge 14 && CONFIG_AC97_BUS="m"
-      CONFIG_SND_VMASTER="y"
-      CONFIG_SND_RAWMIDI="m"
-      CONFIG_SND_AC97_CODEC="m"
-      CONFIG_SND_CS46XX="m"
-    fi
-  fi
-  if alsa_check_kconfig_option "cs46xx-new-dsp"; then
-    if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
-      ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ) &&
-      ( test "$CONFIG_SND_CS46XX" = "y" -o "$CONFIG_SND_CS46XX" = "m" ); then
-      CONFIG_SND_CS46XX_NEW_DSP="y"
-    fi
-  fi
   if alsa_check_kconfig_card "cs5530"; then
     if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
       ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ) &&
@@ -15070,25 +15051,6 @@
       CONFIG_SND_INTEL8X0M="m"
     fi
   fi
-  if alsa_check_kconfig_card "korg1212"; then
-    if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
-      ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ); then
-      CONFIG_SND_TIMER="m"
-      CONFIG_SND_PCM="m"
-      CONFIG_SND_KORG1212="m"
-    fi
-  fi
-  if alsa_check_kconfig_card "maestro3"; then
-    if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
-      ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ); then
-      CONFIG_SND_TIMER="m"
-      CONFIG_SND_PCM="m"
-      test "$kversion.$kpatchlevel" = "2.6" -a $ksublevel -ge 14 && CONFIG_AC97_BUS="m"
-      CONFIG_SND_VMASTER="y"
-      CONFIG_SND_AC97_CODEC="m"
-      CONFIG_SND_MAESTRO3="m"
-    fi
-  fi
   if alsa_check_kconfig_card "mixart"; then
     if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
       ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ); then
@@ -15245,21 +15207,6 @@
       CONFIG_SND_VX222="m"
     fi
   fi
-  if alsa_check_kconfig_card "ymfpci"; then
-    if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
-      ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ); then
-      CONFIG_SND_TIMER="m"
-      CONFIG_SND_HWDEP="m"
-      CONFIG_SND_RAWMIDI="m"
-      CONFIG_SND_PCM="m"
-      test "$kversion.$kpatchlevel" = "2.6" -a $ksublevel -ge 14 && CONFIG_AC97_BUS="m"
-      CONFIG_SND_VMASTER="y"
-      CONFIG_SND_OPL3_LIB="m"
-      CONFIG_SND_MPU401_UART="m"
-      CONFIG_SND_AC97_CODEC="m"
-      CONFIG_SND_YMFPCI="m"
-    fi
-  fi
   if alsa_check_kconfig_card "pdplus"; then
     if  ! ( test "$CONFIG_M68K" = "y" -o "$CONFIG_M68K" = "m" ) &&
       ( test "$CONFIG_SND_PCI" = "y" -o "$CONFIG_SND_PCI" = "m" ); then
