Index: src/libdvdcss.c
===================================================================
--- src/libdvdcss.c	(revision 207)
+++ src/libdvdcss.c	(working copy)
@@ -377,6 +377,10 @@
             return NULL;
         }
     }
+    else
+    {
+        memset( dvdcss->css.p_disc_key, 0, KEY_SIZE );
+    }
 
     /* If the cache is enabled, write the cache directory tag */
     if( psz_cache )
Index: src/css.c
===================================================================
--- src/css.c	(revision 207)
+++ src/css.c	(working copy)
@@ -325,6 +325,7 @@
                                  "cracking title keys instead" );
 
             /* Fallback, but not to DISC as the disc key might be faulty */
+            memset( p_disc_key, 0, KEY_SIZE );
             dvdcss->i_method = DVDCSS_METHOD_TITLE;
             break;
 
