diff -ruN mplayer.eac3/etc/codecs.conf mplayer.eac3.disable_windows_vc1/etc/codecs.conf
--- mplayer.eac3/etc/codecs.conf	2007-10-19 14:13:18.000000000 +0300
+++ mplayer.eac3.disable_windows_vc1/etc/codecs.conf	2007-10-19 14:33:30.000000000 +0300
@@ -805,14 +805,14 @@
   out YUY2,UYVY
 ;  out BGR32,BGR24,BGR16 ;,BGR15
 
-videocodec wmvvc1dmo
-  info "Windows Media Video (VC-1) Advanced Profile Decoder"
-  status working
-  fourcc WVC1
-  driver dmo
-  dll "wvc1dmod.dll"
-  guid 0xc9bfbccf, 0xe60e, 0x4588, 0xa3, 0xdf, 0x5a, 0x03, 0xb1, 0xfd, 0x95, 0x85
-  out YV12,YUY2
+;videocodec wmvvc1dmo
+;  info "Windows Media Video (VC-1) Advanced Profile Decoder"
+;  status working
+;  fourcc WVC1
+;  driver dmo
+;  dll "wvc1dmod.dll"
+;  guid 0xc9bfbccf, 0xe60e, 0x4588, 0xa3, 0xdf, 0x5a, 0x03, 0xb1, 0xfd, 0x95, 0x85
+;  out YV12,YUY2
 
 videocodec wmsdmod
   info "Windows Media Screen Codec 2"
