diff -ruN mplayer.orig/libmpdemux/demux_mpg.c mplayer.hack/libmpdemux/demux_mpg.c
--- mplayer.orig/libmpdemux/demux_mpg.c	2007-10-16 14:31:19.000000000 +0300
+++ mplayer.hack/libmpdemux/demux_mpg.c	2007-10-19 16:52:34.000000000 +0300
@@ -711,7 +711,7 @@
   }
   //FIXME this shouldn't be necessary
   stream_seek(demuxer->stream,tmppos);
-  return file_format;
+  return DEMUXER_TYPE_UNKNOWN;
 }
 
 static int demux_mpg_es_fill_buffer(demuxer_t *demux, demux_stream_t *ds){
