=== modified file 'debian/changelog'
--- debian/changelog	2008-06-12 22:08:24 +0000
+++ debian/changelog	2008-07-22 13:45:50 +0000
@@ -1,3 +1,9 @@
+mythtv (0.22.0~trunk17764-0ubuntu1) intrepid; urgency=low
+
+  * trunk build for upcoming 0.22 
+
+ -- Daniel Kristjansson <danielk@cuymedia.net>  Tue, 22 Jul 2008 09:18:00 -0500
+
 mythtv (0.22.0~trunk17464-0ubuntu1) hardy; urgency=low
 
   * first trunk build for upcoming 0.22 

=== modified file 'debian/control'
--- debian/control	2008-06-12 22:49:46 +0000
+++ debian/control	2008-07-22 13:45:50 +0000
@@ -39,8 +39,6 @@
                 libx264-dev,
                 libxvidcore4-dev,
                 libfftw3-dev | fftw3-dev,
-                libchromexvmc1 [i386 amd64],
-                libchromexvmcpro1 [i386 amd64]
 
 Package: mythtv
 Architecture: all
@@ -148,8 +146,8 @@
 Architecture: any
 Depends: mythtv-common (= ${source:Version}), ${shlibs:Depends}, adduser, zenity | kdebase-bin, gksu | kdebase-bin, ttf-liberation | msttcorefonts, ${misc:Depends}
 Suggests: mythtv-backend, mythmusic, mythweather, mythgallery, mythvideo, mythgame, mythstream, ntp | time-daemon | ntp-simple
-Conflicts: mythtv (<< 0.8-1), mythappearance (<< 0.21.0)
-Replaces: mythtv (<< 0.8-1),  mythappearance (<< 0.21.0)
+Conflicts: mythtv (<< 0.8-1), mythappearance (<< 0.22.0)
+Replaces: mythtv (<< 0.8-1),  mythappearance (<< 0.22.0)
 Description: A personal video recorder application (client)
  MythTV provides a unified graphical interface for recording and viewing
  television programs. Refer to the mythtv package for more information.
@@ -167,7 +165,7 @@
 Package: libmyth-0.22-0
 Architecture: any
 Section: multiverse/libs
-Depends: ${shlibs:Depends}, libqt3-mt-mysql | libqt3c102-mt-mysql
+Depends: ${shlibs:Depends}, libqt4-qt3support, libqt4-sql-mysql
 Conflicts: mythtv (<< 0.7-5)
 Replaces: mythtv (<< 0.7-5)
 Description: Common library code for MythTV and add-on modules (runtime)
@@ -180,9 +178,9 @@
 Package: libmyth-dev
 Architecture: any
 Section: multiverse/libdevel
-Depends: libmyth-0.21-0 (= ${source:Version}), libqt3c102-mt-dev | libqt3-mt-dev
-Provides: libmyth-0.21-0-dev
-Conflicts: libmyth-0.21-dev
+Depends: libmyth-0.22-0 (= ${source:Version}), libqt4-dev
+Provides: libmyth-0.22-0-dev
+Conflicts: libmyth-0.22-dev
 Description: Common library code for MythTV and add-on modules (development)
  MythTV provides a unified graphical interface for recording and viewing
  television programs. Refer to the mythtv package for more information.

=== modified file 'debian/mythtv-backend.files'
--- debian/mythtv-backend.files	2008-03-17 02:16:04 +0000
+++ debian/mythtv-backend.files	2008-07-11 22:30:20 +0000
@@ -2,8 +2,8 @@
 usr/bin/mythcommflag
 usr/bin/mythfilldatabase
 usr/bin/mythjobqueue
-usr/bin/mythprime
-usr/bin/firewire_tester
+#usr/bin/mythprime
+#usr/bin/firewire_tester
 usr/bin/mythtv-setup.real
 usr/share/mythtv/MSRR_scpd.xml
 usr/share/mythtv/CDS_scpd.xml

=== modified file 'debian/mythtv-backend.init'
--- debian/mythtv-backend.init	2008-03-03 07:01:41 +0000
+++ debian/mythtv-backend.init	2008-07-22 13:45:50 +0000
@@ -19,14 +19,14 @@
 
 set -e
 
-prime_firewire()
-{
-    if [ "$ENABLE_FIREWIRE" = "TRUE" ]; then
-	log_daemon_msg "Priming Firewire "
-        su - $USER -c "/usr/bin/mythprime"
-	log_end_msg $?
-    fi
-}
+#prime_firewire()
+#{
+#    if [ "$ENABLE_FIREWIRE" = "TRUE" ]; then
+#	log_daemon_msg "Priming Firewire "
+#        su - $USER -c "/usr/bin/mythprime"
+#	log_end_msg $?
+#    fi
+#}
 
 USER=mythtv
 USER_HOME=$(grep ^$USER /etc/passwd | awk -F : '{print $6}')
@@ -66,7 +66,7 @@
 			rm -f $RUNDIR/$NAME.pid
 		fi
 	fi
-	prime_firewire
+	#prime_firewire
 	log_daemon_msg "Starting $DESC: $NAME "
 	start-stop-daemon --start --pidfile $RUNDIR/$NAME.pid \
 		--chuid $USER --nicelevel $NICE --exec $DAEMON -- $ARGS
@@ -84,7 +84,7 @@
 	start-stop-daemon --stop --oknodo --pidfile $RUNDIR/$NAME.pid \
                 --chuid $USER --exec $DAEMON -- $ARGS
 	sleep 3
-	prime_firewire
+	#prime_firewire
 	start-stop-daemon --start --pidfile $RUNDIR/$NAME.pid \
                 --chuid $USER --nicelevel $NICE --exec $DAEMON -- $ARGS
     log_end_msg $?

=== modified file 'debian/rules'
--- debian/rules	2008-06-12 22:49:46 +0000
+++ debian/rules	2008-07-12 00:29:21 +0000
@@ -7,7 +7,7 @@
 SVN_TYPE+=trunk
 SVN_MAJOR_RELEASE+=22
 SVN_MINOR_RELEASE+=0
-SVN_REVISION+=17463
+SVN_REVISION+=17764
 #SVN_BRANCH+= http://svn.mythtv.org/svn/branches/release-0-$(SVN_MAJOR_RELEASE)-$(SVN_TYPE)/$(SVN_PACKAGE)
 SVN_BRANCH+= http://svn.mythtv.org/svn/$(SVN_TYPE)/$(SVN_PACKAGE)
 GREATER+=false
@@ -32,6 +32,9 @@
 	tar czf $(CURDIR)/../$(TARFILE) $(SVN_PACKAGE)-$(SVN_RELEASE)$(SUFFIX)
 	rm -rf $(CURDIR)/$(SVN_PACKAGE)-$(SVN_RELEASE)$(SUFFIX)
 
+extract-source:
+	tar xvf mythtv-$(SVN_TYPE)-r$(SVN_REVISION).tbz
+
 NCPUS := $(shell getconf _NPROCESSORS_ONLN 2>/dev/null || echo 1)
 
 ifeq ($(NCPUS),-1)
@@ -75,12 +78,14 @@
 endif
 endif
 
+patch-stamp: extract-source patch
+
 configure: configure-stamp
 configure-stamp: patch-stamp
 	dh_testdir
 
 	#give us firewire testing support (from contrib)
-	ln -sf ../../contrib/development/firewire_tester/firewire_tester.c programs/firewire_tester
+#	ln -sf ../../contrib/development/firewire_tester/firewire_tester.c programs/firewire_tester
 	
 	#update SVN REVISION to show the right version
 	sed -i -e "s/\`(svnversion \$$\$${SVNTREEDIR} 2>\/dev\/null) || echo Unknown\`/$(SVN_REVISION)/" version.pro
@@ -89,7 +94,7 @@
 	--enable-audio-oss --enable-audio-jack --enable-audio-arts --enable-dvb \
 	--enable-ivtv --enable-firewire --enable-joystick-menu \
 	--enable-opengl-vsync --with-bindings=perl --enable-opengl-video \
-	--enable-ffmpeg-pthreads\
+	--enable-ffmpeg-pthreads --qmake=qmake-qt4 \
         $(CONFIGURE_OPTS) $(TOP_SECRET_OPTS)
 
 	touch configure-stamp
@@ -98,7 +103,7 @@
 build-stamp:
 	dh_testdir
 
-	$(CCVARS) qmake -o Makefile PREFIX=/usr mythtv.pro
+	$(CCVARS) qmake-qt4 -o Makefile PREFIX=/usr mythtv.pro
 
 	$(MAKE) -j $(NCPUS) $(CCVARS)
 
@@ -117,12 +122,16 @@
 	rm -rf debian/shlibs.local
 
 	#this is a symlink manually added in
-	rm -f programs/firewire_tester.c
+#	rm -f programs/firewire_tester.c
 
-	debian/rules unpatch
+#	debian/rules unpatch
 
 	dh_clean Makefile config.log setup/Makefile programs/mythbackend/version.cpp \
-	programs/mythfrontend/version.cpp
+		programs/mythfrontend/version.cpp
+	rm -rf	AUTHORS COPYING FAQ Makefile README UPGRADING bindings config \
+		config.err config.h config.log config.mak configure configure-stamp \
+		contrib database docs filters i18n keys.txt libs mythtv.pro \
+		patch-stamp programs settings.pro themes version.pro version.sh
 
 install: build
 	dh_testdir
@@ -206,7 +215,7 @@
 	dh_testroot -i
 	dh_installdebconf -i
 	dh_installdocs -A README debian/README.Debian AUTHORS FAQ UPGRADING keys.txt
-	dh_installdocs -pmythtv-backend contrib/development/firewire_tester/README 
+#	dh_installdocs -pmythtv-backend contrib/development/firewire_tester/README 
 	dh_installdocs -pmythtv-doc docs/mythtv-HOWTO*
 	dh_installexamples -pmythtv-doc contrib/config_files/*
 	dh_installcron -i
@@ -242,7 +251,7 @@
 	dh_makeshlibs -a -V -Xusr/lib/mythtv/filters
 	dh_installdeb -a
 #	dh_shlibdeps -a -ldebian/libmyth-0.19/usr/lib
-	dh_shlibdeps -a
+#	dh_shlibdeps -a
 	dh_gencontrol -a
 	dh_md5sums -a
 	dh_builddeb -a

