diff -ur libsdl1.2-1.2.10.orig/src/video/Xext/Xxf86dga/XF86DGA.c libsdl1.2-1.2.10/src/video/Xext/Xxf86dga/XF86DGA.c
--- libsdl1.2-1.2.10.orig/src/video/Xext/Xxf86dga/XF86DGA.c	2006-05-01 18:02:47.000000000 +1000
+++ libsdl1.2-1.2.10/src/video/Xext/Xxf86dga/XF86DGA.c	2006-10-21 12:59:32.000000000 +1000
@@ -18,7 +18,6 @@
 #define HAS_MMAP_ANON
 #include <sys/types.h>
 #include <sys/mman.h>
-#include <asm/page.h>   /* PAGE_SIZE */
 #define HAS_SC_PAGESIZE /* _SC_PAGESIZE may be an enum for Linux */
 #define HAS_GETPAGESIZE
 #endif /* linux */
diff -ur libsdl1.2-1.2.10.orig/src/video/Xext/Xxf86dga/XF86DGA2.c libsdl1.2-1.2.10/src/video/Xext/Xxf86dga/XF86DGA2.c
--- libsdl1.2-1.2.10.orig/src/video/Xext/Xxf86dga/XF86DGA2.c	2006-05-01 18:02:47.000000000 +1000
+++ libsdl1.2-1.2.10/src/video/Xext/Xxf86dga/XF86DGA2.c	2006-10-21 13:02:43.000000000 +1000
@@ -947,7 +947,7 @@
         }
         /* Warning: On PPC, the size and virtual need to be offset by:
            (((long)finfo.smem_start) -
-           (((long)finfo.smem_start)&~(PAGE_SIZE-1)))
+           (((long)finfo.smem_start)&~(sysconf(_SC_PAGESIZE)-1)))
          */
         base = 0;
         size = pMap->size = finfo.smem_len;
diff -ur libsdl1.2-1.2.10.orig/src/video/fbcon/SDL_fbvideo.c libsdl1.2-1.2.10/src/video/fbcon/SDL_fbvideo.c
--- libsdl1.2-1.2.10.orig/src/video/fbcon/SDL_fbvideo.c	2006-05-17 13:16:07.000000000 +1000
+++ libsdl1.2-1.2.10/src/video/fbcon/SDL_fbvideo.c	2006-10-21 13:01:23.000000000 +1000
@@ -29,7 +29,6 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
-#include <asm/page.h>		/* For definition of PAGE_SIZE */
 #include <linux/vt.h>
 
 #include "SDL_video.h"
@@ -547,7 +546,7 @@
 
 	/* Memory map the device, compensating for buggy PPC mmap() */
 	mapped_offset = (((long)finfo.smem_start) -
-	                (((long)finfo.smem_start)&~(PAGE_SIZE-1)));
+	                (((long)finfo.smem_start)&~(sysconf(_SC_PAGESIZE)-1)));
 	mapped_memlen = finfo.smem_len+mapped_offset;
 	mapped_mem = do_mmap(NULL, mapped_memlen,
 	                  PROT_READ|PROT_WRITE, MAP_SHARED, console_fd, 0);
diff -ur libsdl1.2-1.2.10.orig/src/video/ps2gs/SDL_gsyuv.c libsdl1.2-1.2.10/src/video/ps2gs/SDL_gsyuv.c
--- libsdl1.2-1.2.10.orig/src/video/ps2gs/SDL_gsyuv.c	2006-05-17 13:16:07.000000000 +1000
+++ libsdl1.2-1.2.10/src/video/ps2gs/SDL_gsyuv.c	2006-10-21 13:00:53.000000000 +1000
@@ -27,7 +27,6 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
-#include <asm/page.h>		/* For definition of PAGE_SIZE */
 
 #include "SDL_video.h"
 #include "SDL_gsyuv_c.h"
@@ -180,7 +179,7 @@
 
 	/* Allocate a DMA area for pixel conversion */
 	bpp = this->screen->format->BytesPerPixel;
-	map_offset = (mapped_len + (PAGE_SIZE - 1)) & ~(PAGE_SIZE - 1);
+	map_offset = (mapped_len + (sysconf(_SC_PAGESIZE) - 1)) & ~(sysconf(_SC_PAGESIZE) - 1);
 	hwdata->dma_len = hwdata->macroblocks * (16 * 16 + 8 * 8 + 8 * 8) +
 	                  width * height * bpp +
 	                  hwdata->macroblocks * (16 * sizeof(long long)) +
