diff -ruN hal-0.5.0-old/README hal-0.5.0/README
--- hal-0.5.0-old/README	2004-12-14 03:47:58.000000000 +0100
+++ hal-0.5.0/README	2005-04-13 12:32:18.448771192 +0200
@@ -2,8 +2,6 @@
 HAL is a hardware abstraction layer
 
 
-See also the file HACKING for notes of interest to developers working on HAL.
-
 See http://www.freedesktop.org/Software/hal for lots of documentation, 
 mailing lists, etc.
 
