Index: hal-0.5.14/hald/linux/addons/addon-storage.c
===================================================================
--- hal-0.5.14.orig/hald/linux/addons/addon-storage.c	2013-04-21 17:27:18.000000000 -0400
+++ hal-0.5.14/hald/linux/addons/addon-storage.c	2013-04-21 17:29:31.918863034 -0400
@@ -32,6 +32,7 @@
 #include <linux/cdrom.h>
 #include <linux/fs.h>
 #include <mntent.h>
+#include <sys/types.h>
 #include <scsi/sg.h>
 #include <stdint.h>
 #include <stdio.h>
