--- create_config.sh.old	2005-12-02 22:19:48.000000000 +0200
+++ create_config.sh	2006-01-23 16:01:07.000000000 +0200
@@ -38,21 +38,6 @@
 echo "#define CHAN_SCCP_CONFIG_H" >>$CONFIGFILE
 echo >>$CONFIGFILE
 
-echo -n "Build PARK functions (y/n)[n]?"
-read key
-if [ "$key" = "y" ]
-then
-	echo "#define CS_SCCP_PARK"  >>$CONFIGFILE
-fi
-
-echo -n "Build PICKUP functions (y/n)[n]?"
-read key
-if [ "$key" = "y" ]
-then
-        echo "#define CS_SCCP_PICKUP"  >>$CONFIGFILE
-fi
-
-
 if grep -q "struct ast_channel_tech" $INCLUDEDIR/channel.h; then
 	echo "#define CS_AST_HAS_TECH_PVT" >>$CONFIGFILE
 	echo " * found 'struct ast_channel_tech'"
