diff -Naur xine-lib-1.1.1.old/configure.ac xine-lib-1.1.1/configure.ac
--- xine-lib-1.1.1.old/configure.ac	2005-11-14 13:34:45.000000000 +0100
+++ xine-lib-1.1.1/configure.ac	2006-01-20 17:59:12.401224336 +0100
@@ -2242,21 +2242,8 @@
 src/input/vcd/libvcd/libvcd/Makefile
 src/liba52/Makefile
 src/libdts/Makefile
-src/libfaad/Makefile
-src/libfaad/codebook/Makefile
-src/libffmpeg/Makefile
-src/libffmpeg/libavcodec/Makefile
-src/libffmpeg/libavcodec/armv4l/Makefile
-src/libffmpeg/libavcodec/i386/Makefile
-src/libffmpeg/libavcodec/mlib/Makefile
-src/libffmpeg/libavcodec/alpha/Makefile
-src/libffmpeg/libavcodec/ppc/Makefile
-src/libffmpeg/libavcodec/sparc/Makefile
-src/libffmpeg/libavcodec/libpostproc/Makefile
-src/libffmpeg/libavutil/Makefile
 src/libflac/Makefile
 src/liblpcm/Makefile
-src/libmad/Makefile
 src/libmpeg2/Makefile
 src/libmusepack/Makefile
 src/libmusepack/musepack/Makefile
@@ -2281,6 +2268,7 @@
 src/libreal/Makefile
 src/post/Makefile
 src/post/planar/Makefile
+src/libpostproc/Makefile
 src/post/goom/Makefile
 src/post/mosaico/Makefile
 src/post/visualizations/Makefile
diff -Naur xine-lib-1.1.1.old/src/libpostproc/fastmemcpy.h xine-lib-1.1.1/src/libpostproc/fastmemcpy.h
--- xine-lib-1.1.1.old/src/libpostproc/fastmemcpy.h	1970-01-01 01:00:00.000000000 +0100
+++ xine-lib-1.1.1/src/libpostproc/fastmemcpy.h	2006-01-20 17:58:14.594012368 +0100
@@ -0,0 +1,4 @@
+#if 0
+extern void *xine_fast_memcpy(void *to, const void *from, size_t len);
+#define memcpy(a,b,c) xine_fast_memcpy(a,b,c)
+#endif
diff -Naur xine-lib-1.1.1.old/src/libpostproc/Makefile.am xine-lib-1.1.1/src/libpostproc/Makefile.am
--- xine-lib-1.1.1.old/src/libpostproc/Makefile.am	2005-10-30 23:37:45.000000000 +0100
+++ xine-lib-1.1.1/src/libpostproc/Makefile.am	2006-01-20 17:58:14.594012368 +0100
@@ -7,7 +7,7 @@
 
 noinst_LTLIBRARIES = libpostprocess.la
 
-EXTRA_DIST = postprocess_template.c postprocess_altivec_template.c
+EXTRA_DIST = postprocess_template.c postprocess_altivec_template.c fastmemcpy.h
 
 libpostprocess_la_SOURCES = postprocess.c
 
diff -Naur xine-lib-1.1.1.old/src/Makefile.am xine-lib-1.1.1/src/Makefile.am
--- xine-lib-1.1.1.old/src/Makefile.am	2005-01-14 16:24:08.000000000 +0100
+++ xine-lib-1.1.1/src/Makefile.am	2006-01-20 17:58:25.944286864 +0100
@@ -8,8 +8,8 @@
 	video_out \
 	dxr3 \
 	input \
+	libpostproc \
 	demuxers \
-	libffmpeg \
 	libmpeg2 \
 	liba52 \
 	libspudec \
@@ -18,7 +18,6 @@
 	libspudvb \
 	libsputext \
 	libdts \
-	libmad \
 	libw32dll \
 	liblpcm \
 	libxinevdec \
@@ -27,7 +26,6 @@
 	libtheora \
 	libspeex \
 	libreal \
-	libfaad \
 	libflac \
         libmusepack \
 	post
diff -Naur xine-lib-1.1.1.old/src/post/deinterlace/plugins/Makefile.am xine-lib-1.1.1/src/post/deinterlace/plugins/Makefile.am
--- xine-lib-1.1.1.old/src/post/deinterlace/plugins/Makefile.am	2005-06-05 18:00:06.000000000 +0200
+++ xine-lib-1.1.1/src/post/deinterlace/plugins/Makefile.am	2006-01-20 17:58:14.595012216 +0100
@@ -30,7 +30,7 @@
 
 # libpostproc is here so we can use their nice mangle.h
 AM_CFLAGS = -I$(top_srcdir)/src/post/deinterlace \
-	-I$(top_srcdir)/src/libffmpeg/libavcodec/libpostproc
+	-I$(top_srcdir)/src/libpostproc
 
 libdir = $(XINE_PLUGINDIR)/post
 
diff -Naur xine-lib-1.1.1.old/src/post/planar/Makefile.am xine-lib-1.1.1/src/post/planar/Makefile.am
--- xine-lib-1.1.1.old/src/post/planar/Makefile.am	2005-10-30 23:37:45.000000000 +0100
+++ xine-lib-1.1.1/src/post/planar/Makefile.am	2006-01-20 17:58:14.596012064 +0100
@@ -1,12 +1,12 @@
 include $(top_srcdir)/misc/Makefile.common
 
-POSTPROC_INT_LIB = $(top_builddir)/src/libffmpeg/libavcodec/libpostproc/libpostprocess.la
+POSTPROC_INT_LIB = $(top_builddir)/src/libpostproc/libpostprocess.la
 
 if HAVE_FFMPEG
 postproc_lib = $(FFMPEG_POSTPROC_LIBS)
 ff_cppflags = $(FFMPEG_CPPFLAGS)
 else
-ff_cppflags = -I$(top_srcdir)/src/libffmpeg/libavcodec/libpostproc
+ff_cppflags = -I$(top_srcdir)/src/libpostproc
 postproc_lib = $(POSTPROC_INT_LIB)
 postproc_dep = $(postproc_lib)
 endif
@@ -26,4 +26,4 @@
 xineplug_post_planar_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
 $(POSTPROC_INT_LIB):
-	cd $(top_builddir)/src/libffmpeg/libavcodec/libpostproc && $(MAKE) libpostprocess.la
+	cd $(top_builddir)/src/libpostproc && $(MAKE) libpostprocess.la
