diff -Nru arghdirector-0.2.6b/arghdirector.c arghdirector-0.2.6b_patched/arghdirector.c
--- arghdirector-0.2.6b/arghdirector.c	2007-02-11 20:30:07.000000000 +0100
+++ arghdirector-0.2.6b_patched/arghdirector.c	2008-06-02 23:16:12.998895526 +0200
@@ -57,9 +57,9 @@
 {
   const cChannel* Channel = Channels.GetByNumber(Device->CurrentChannel());
 #if VDRVERSNUM < 10333
-	if(Channel->GetLinkChannels() != NULL && Channel->GetLinkChannels()->Count() > 1)
+	if(Channel && Channel->GetLinkChannels() != NULL && Channel->GetLinkChannels()->Count() > 1)
 #else
-	if(Channel->LinkChannels() != NULL && Channel->LinkChannels()->Count() > 1)
+	if(Channel && Channel->LinkChannels() != NULL && Channel->LinkChannels()->Count() > 1)
 #endif
 		if(parent && ChannelNumber != 0)
 		{
diff -Nru arghdirector-0.2.6b/directorosd.c arghdirector-0.2.6b_patched/directorosd.c
--- arghdirector-0.2.6b/directorosd.c	2007-11-04 16:53:47.000000000 +0100
+++ arghdirector-0.2.6b_patched/directorosd.c	2008-06-02 23:16:13.014896437 +0200
@@ -30,13 +30,9 @@
 	Channel = Channels.GetByNumber(cDevice::PrimaryDevice()->CurrentChannel());
 	cChannel* initChannel = Channel;
 
-#if VDRVERSNUM < 10333
-	if (!Channel->GetLinkChannels())
-		Channel = Channel->GetRefChannel();
-#else
-	if (!Channel->LinkChannels())
+
+	if (Channel && !Channel->LinkChannels())
 		Channel = (cChannel*)Channel->RefChannel();
-#endif
 
 	if (!Channel)
 		Channel = initChannel;
@@ -50,11 +46,8 @@
 	if (Channel)
 	{
 
-#if VDRVERSNUM < 10333
-		cLinkChannels* linkChannels = Channel->GetLinkChannels();
-#else
+
 		cLinkChannels* linkChannels = (cLinkChannels*)Channel->LinkChannels();
-#endif
 
 		if (linkChannels)
       {
@@ -197,15 +190,17 @@
 		current--;
 	
 	cChannel *newChannel =  getChannel(current);
-	free(channelname);
-	asprintf(&channelname, "%s", newChannel->Name());
+	if (newChannel)
+		{
+		free(channelname);
+		asprintf(&channelname, "%s", newChannel->Name());
 	
-	cDevice::PrimaryDevice()->SwitchChannel(newChannel, true);
-
-	if (isOsdShowing)
-		drawOsd();
-	else if (mData->mDisplayChannelInfo == 1)
-		Skins.Message(mtInfo, channelname);
+		cDevice::PrimaryDevice()->SwitchChannel(newChannel, true);
+		if (isOsdShowing)
+			drawOsd();
+		else if (mData->mDisplayChannelInfo == 1)
+			Skins.Message(mtInfo, channelname);
+		}
 }
 
 void cDirectorOsd::ChannelUp()
@@ -223,17 +218,18 @@
 	if (lastChannel == current)
 		return;
 	
-	cChannel *newChannel = getChannel(current);
-	
-	free(channelname);	
-	asprintf(&channelname, "%s", newChannel->Name());
+	cChannel *newChannel = getChannel(current);	
+	if (newChannel)
+		{
+		free(channelname);
+		asprintf(&channelname, "%s", newChannel->Name());
 	
-	cDevice::PrimaryDevice()->SwitchChannel(newChannel, true);
-
-	if (isOsdShowing)
-		drawOsd();
-	else if (mData->mDisplayChannelInfo == 1)
-		Skins.Message(mtInfo, channelname);
+		cDevice::PrimaryDevice()->SwitchChannel(newChannel, true);
+		if (isOsdShowing)
+			drawOsd();
+		else if (mData->mDisplayChannelInfo == 1)
+			Skins.Message(mtInfo, channelname);
+		}
 }
 
 void cDirectorOsd::SwitchtoChannel(int channelnumber)
@@ -251,16 +247,18 @@
 	//fprintf(stderr, "current %d\n", current);
 	
 	cChannel *newChannel = getChannel(current);
+	if (newChannel)
+		{	
+		free(channelname);
+		asprintf(&channelname, "%s", newChannel->Name());
 	
-	free(channelname);
-	asprintf(&channelname, "%s", newChannel->Name());
-	
-	cDevice::PrimaryDevice()->SwitchChannel(newChannel, true);
+		cDevice::PrimaryDevice()->SwitchChannel(newChannel, true);
 
-	if (isOsdShowing)
-		drawOsd();
-	else
-		Skins.Message(mtInfo, channelname);
+		if (isOsdShowing)
+			drawOsd();
+		else
+			Skins.Message(mtInfo, channelname);
+		}
 }
 
 void cDirectorOsd::CursorOK(){}
@@ -275,27 +273,26 @@
 		return Channel;
 	else
 	{
-#if VDRVERSNUM < 10333
-		cLinkChannels* linkChannels = Channel->GetLinkChannels();
-#else
-		cLinkChannels* linkChannels = (cLinkChannels*)Channel->LinkChannels();
-#endif
-		cLinkChannel* lca = linkChannels->First();
 
-		for (int i = 1; i < channelnumber; i++)
-			lca = linkChannels->Next(lca);
+		cLinkChannels* linkChannels = Channel?(cLinkChannels*)Channel->LinkChannels():NULL;
+
+		if (linkChannels)
+		{
+			cLinkChannel* lca = linkChannels->First();
 
-		return lca->Channel();
+			for (int i = 1; i < channelnumber; i++)
+				lca = linkChannels->Next(lca);
+
+			return lca->Channel();
+		}
+		else return NULL;
 	}
 }
 
 int cDirectorOsd::getNumLinks()
 {
-#if VDRVERSNUM < 10333
-	cLinkChannels* linkChannels = Channel->GetLinkChannels();
-#else
-	cLinkChannels* linkChannels = (cLinkChannels*)Channel->LinkChannels();
-#endif
+	cLinkChannels* linkChannels = Channel?(cLinkChannels*)Channel->LinkChannels():NULL;
+
 	if (linkChannels)
 		return linkChannels->Count();
 	else
@@ -327,18 +324,16 @@
 	
 	if (Channel)
 	{
-#if VDRVERSNUM < 10333
-		cLinkChannels* linkChannels = Channel->GetLinkChannels();
-#else
+
 		cLinkChannels* linkChannels = (cLinkChannels*)Channel->LinkChannels();
-#endif
+
 		if (linkChannels)
 			number = getNumLinks() + 1;
 		else
 			number = 1;
 		
 		char *buffer=0;
-		asprintf(&buffer,"1 %s %d", Channel->Name(), number);
+		asprintf(&buffer,"1 %s %d", Channel?Channel->Name():"NULL", number);
 				
 		int theHeigth = font->Height("A");
 		int m_height = (number * theHeigth) + 10;
@@ -350,7 +345,7 @@
 		for (int i = 0; i < number; i++)
 		{
 			cChannel *newChannel =  getChannel(i);
-			asprintf(&buffer,"%d %s", i + 1, newChannel->Name());
+			asprintf(&buffer,"%d %s", i + 1, newChannel?newChannel->Name():"NULL");
 			int width = font->Width(buffer);
 			if(width > gr_width)
 				gr_width = width;
@@ -361,7 +356,7 @@
 		if (gr_width%2 == 0)
 			gr_width++;
 		
-		asprintf(&buffer,"1 %s %d", Channel->Name(), number);
+		asprintf(&buffer,"1 %s %d", Channel?Channel->Name():"NULL", number);
   
   		if (!osd)
   			osd = cOsdProvider::NewOsd(100, 50);
@@ -386,18 +381,18 @@
 				if (mData->mPortalmode == 0)
 				{
 					if(i < 9)
-						asprintf(&buffer,"%d   %s", i + 1, newChannel->Name());
+						asprintf(&buffer,"%d   %s", i + 1, newChannel?newChannel->Name():"NULL");
 					else
-						asprintf(&buffer,"%d %s", i + 1, newChannel->Name());
+						asprintf(&buffer,"%d %s", i + 1, newChannel?newChannel->Name():"NULL");
 				}
 				else
 				{
 					if (i == 0)
-                  asprintf(&buffer,"     %s", newChannel->Name());
+                  asprintf(&buffer,"     %s", newChannel?newChannel->Name():"NULL");
 					else if (i < 10)
-						asprintf(&buffer,"%d   %s", i, newChannel->Name());
+						asprintf(&buffer,"%d   %s", i, newChannel?newChannel->Name():"NULL");
 					else
-						asprintf(&buffer,"%d %s", i, newChannel->Name());
+						asprintf(&buffer,"%d %s", i, newChannel?newChannel->Name():"NULL");
 				}
 				if (current == i)
 					osd->DrawText(7, y, buffer, clrBlack, bgColor, font);
diff -Nru arghdirector-0.2.6b/menudirector.c arghdirector-0.2.6b_patched/menudirector.c
--- arghdirector-0.2.6b/menudirector.c	2007-11-17 11:14:20.000000000 +0100
+++ arghdirector-0.2.6b_patched/menudirector.c	2008-06-02 23:16:13.042898033 +0200
@@ -52,7 +52,7 @@
    {
       if (!mNext)
       {
-         if (strlen(p) >= 3 && strstr(p, "D ") == p && isdigit(p[2]))
+         if (p && strlen(p) >= 3 && strstr(p, "D ") == p && isdigit(p[2]))
             asprintf(&chName, "%s%.*s", mMenu->hotKey(""), 3, p);
          else
             asprintf(&chName, "%s%s", mMenu->hotKey(""), p);
@@ -62,8 +62,9 @@
       {
          char r = mEvent->SeenWithin(30) && mEvent->IsRunning() ? 251 : ' ';  // WarEagle
          char t = TimerMatchChars[match];
-         int p = (int)roundf((float)(time(0) - mEvent->StartTime()) 
-                                    / (float)(mEvent->Duration()) * 6.0);
+         int p = 0;
+         if (mEvent->Duration() != 0) /* avoid div by zero */
+            p = (int)roundf((float)(time(0) - mEvent->StartTime()) / (float)(mEvent->Duration()) * 6.0);              
 
          if      (p < 0)  p = 0;
          else if (p > 6)  p = 6;
@@ -126,9 +127,12 @@
 
 void cMenuDirector::ChannelSwitchNonOsd(const cChannel* channel)
 {
-  cDevice::PrimaryDevice()->SwitchChannel(channel, true);
-  if ((mData) && (mData->mDisplayChannelInfo == 1))
-    Skins.Message(mtInfo,channel->Name());
+  if (channel)
+    {
+    cDevice::PrimaryDevice()->SwitchChannel(channel, true);
+    if ((mData) && (mData->mDisplayChannelInfo == 1))
+      Skins.Message(mtInfo,channel->Name());
+    }
 }
 
 void cMenuDirector::ChannelSwitchNonOsd(int direction)
@@ -142,43 +146,44 @@
   if(GetLinkChannelsOutOfChannel(mReferenceChannel))
   {
     cChannel* currentChannel = Channels.GetByNumber(cDevice::PrimaryDevice()->CurrentChannel());
+ 
     const cLinkChannels* linkChannels = GetLinkChannelsOutOfChannel(mReferenceChannel);
 
     if ((currentChannel == mReferenceChannel) && (direction == -1))
     {
-      ChannelSwitchNonOsd(linkChannels->Last()->Channel());
+      if (linkChannels) ChannelSwitchNonOsd(linkChannels->Last()->Channel());
       return;
     }
     if ((currentChannel == mReferenceChannel) && (direction == +1))
     {
-      ChannelSwitchNonOsd(linkChannels->First()->Channel());
+      if (linkChannels) ChannelSwitchNonOsd(linkChannels->First()->Channel());
       return;
     }
     
     const cChannel* prevChannel = mReferenceChannel;
-
-    for (cLinkChannel* linkChannel = linkChannels->First(); linkChannel; 
+    if (linkChannels)
+      for (cLinkChannel* linkChannel = linkChannels->First(); linkChannel; 
          linkChannel = linkChannels->Next(linkChannel))
-    {
-      if (linkChannel->Channel() == currentChannel)
       {
-        if (direction == -1)
+        if (linkChannel->Channel() == currentChannel)
         {
-          ChannelSwitchNonOsd(prevChannel);
-          return;
-        }
-        else
-        {
-          if (linkChannels->Next(linkChannel) == NULL)
-            ChannelSwitchNonOsd(mReferenceChannel);
+          if (direction == -1)
+          {
+            ChannelSwitchNonOsd(prevChannel);
+            return;
+          }
           else
-            ChannelSwitchNonOsd(linkChannels->Next(linkChannel)->Channel());
-          return;
+          {
+            if (linkChannels->Next(linkChannel) == NULL)
+              ChannelSwitchNonOsd(mReferenceChannel);
+            else
+              ChannelSwitchNonOsd(linkChannels->Next(linkChannel)->Channel());
+            return;
+          }
         }
-      }
 
-      prevChannel = linkChannel->Channel();
-    }
+        prevChannel = linkChannel->Channel();
+      }
   }
 }
 
@@ -338,7 +343,7 @@
          const cLinkChannels* linkChannels = GetLinkChannelsOutOfChannel(mReferenceChannel);
        
          dsyslog("Updating director menu with (%d) linked channels", linkChannels->Count());
-
+         if (linkChannels)
          for (cLinkChannel* linkChannel = linkChannels->First(); linkChannel; 
               linkChannel = linkChannels->Next(linkChannel))
          {
