Index: mythplugins-0.22/mythmusic/mythmusic/globalsettings.cpp
===================================================================
--- mythplugins-0.22.orig/mythmusic/mythmusic/globalsettings.cpp	2009-11-11 17:10:07.000000000 +0100
+++ mythplugins-0.22/mythmusic/mythmusic/globalsettings.cpp	2009-11-11 18:49:57.000000000 +0100
@@ -40,7 +40,7 @@
 #ifdef Q_WS_MACX
     gc->setValue(QDir::homePath() + "/Music");
 #else
-    gc->setValue("/mnt/store/music/");
+    gc->setValue("/var/lib/music/");
 #endif
     gc->setHelpText(QObject::tr("This directory must exist, and the user "
                     "running MythMusic needs to have write permission "
