Index: mythplugins-0.22/mythweb/mythweb.conf.apache
===================================================================
--- mythplugins-0.22.orig/mythweb/mythweb.conf.apache	2008-10-22 20:50:54.000000000 +0200
+++ mythplugins-0.22/mythweb/mythweb.conf.apache	2009-11-11 18:51:48.000000000 +0100
@@ -26,10 +26,10 @@
 #    /var/www/html/mythweb
 #    /srv/www/htdocs/mythweb
 #
-    <Directory "/var/www/html/data">
+    <Directory "/var/www/mythweb/data">
         Options -All +FollowSymLinks +IncludesNoExec
     </Directory>
-    <Directory "/var/www/html" >
+    <Directory "/var/www/mythweb" >
 
     ############################################################################
     # I *strongly* urge you to turn on authentication for MythWeb.  It is disabled
