Index: mythplugins-0.22+fixes20100105/mythmusic/mythmusic/playlist.cpp
===================================================================
--- mythplugins-0.22+fixes20100105.orig/mythmusic/mythmusic/playlist.cpp	2010-01-18 09:38:30.000000000 +0100
+++ mythplugins-0.22+fixes20100105/mythmusic/mythmusic/playlist.cpp	2010-01-18 09:39:15.000000000 +0100
@@ -1289,7 +1289,7 @@
                                       100);
     progress->setProgress(1);
 
-    QStringList args("mkisofs");
+    QStringList args("genisoimage");
     args += "-graft-points";
     args += "-path-list";
     args += tmprecordlist;
@@ -1326,7 +1326,7 @@
             {
                 if (!isofs.normalExit())
                 {
-                    VERBOSE(VB_IMPORTANT, "Unable to run 'mkisofs'");
+                    VERBOSE(VB_IMPORTANT, "Unable to run 'genisoimage'");
                     retval = 1;
                 }
                 break;
@@ -1335,7 +1335,7 @@
     }
     else
     {
-        VERBOSE(VB_IMPORTANT, "Unable to run 'mkisofs'");
+        VERBOSE(VB_IMPORTANT, "Unable to run 'genisoimage'");
         retval = 1;
     }
 
Index: mythplugins-0.22+fixes20100105/mytharchive/mytharchivehelper/main.cpp
===================================================================
--- mythplugins-0.22+fixes20100105.orig/mytharchive/mytharchivehelper/main.cpp	2010-01-18 09:39:34.000000000 +0100
+++ mythplugins-0.22+fixes20100105/mytharchive/mytharchivehelper/main.cpp	2010-01-18 09:42:53.000000000 +0100
@@ -194,8 +194,8 @@
 
     tempDirectory += "work/";
 
-    QString mkisofs = gContext->GetSetting("MythArchiveMkisofsCmd", "mkisofs");
-    QString command = mkisofs + " -R -J -V 'MythTV Archive' -o ";
+    QString genisoimage = gContext->GetSetting("MythArchiveMkisofsCmd", "genisoimage");
+    QString command = genisoimage + " -R -J -V 'MythTV Archive' -o ";
     command += tempDirectory + "mythburn.iso " + sourceDirectory;
 
     int res = system(qPrintable(command));
@@ -203,7 +203,7 @@
         res = WEXITSTATUS(res);
     if (res != 0)
     {
-        VERBOSE(VB_JOBQUEUE, QString("ERROR: Failed while running mkisofs. Result: %1")
+        VERBOSE(VB_JOBQUEUE, QString("ERROR: Failed while running genisoimage. Result: %1")
                 .arg(res));
         return false;
     }
Index: mythplugins-0.22+fixes20100105/mytharchive/mythburn/scripts/mythburn.py
===================================================================
--- mythplugins-0.22+fixes20100105.orig/mytharchive/mythburn/scripts/mythburn.py	2010-01-18 09:40:21.000000000 +0100
+++ mythplugins-0.22+fixes20100105/mytharchive/mythburn/scripts/mythburn.py	2010-01-18 09:43:59.000000000 +0100
@@ -2567,7 +2567,7 @@
 def CreateDVDISO(title):
     write("Creating ISO image")
     checkCancelFlag()
-    command = path_mkisofs[0] + ' -dvd-video '
+    command = path_genisoimage[0] + ' -dvd-video '
     command += ' -V ' + quoteFilename(title)
     command += ' -o ' + os.path.join(getTempPath(), 'mythburn.iso')
     command += " " + os.path.join(getTempPath(),'dvd')
@@ -2575,7 +2575,7 @@
     result = runCommand(command)
 
     if result<>0:
-        fatalError("Failed while running mkisofs.\n"
+        fatalError("Failed while running genisoimage.\n"
                    "Command was %s" % command)
 
     write("Finished creating ISO image")
@@ -5270,7 +5270,7 @@
     global preferredlang2, useFIFO, encodetoac3, alwaysRunMythtranscode
     global copyremoteFiles, mainmenuAspectRatio, chaptermenuAspectRatio, dateformat
     global timeformat, clearArchiveTable, nicelevel, drivespeed, path_mplex, path_ffmpeg
-    global path_dvdauthor, path_mkisofs, path_growisofs, path_tcrequant, addSubtitles
+    global path_dvdauthor, path_genisoimage, path_growisofs, path_tcrequant, addSubtitles
     global path_jpeg2yuv, path_spumux, path_mpeg2enc, path_projectx, useprojectx, progresslog
     global progressfile, jobfile
 
@@ -5364,7 +5364,7 @@
     path_mplex = [defaultsettings["MythArchiveMplexCmd"], os.path.split(defaultsettings["MythArchiveMplexCmd"])[1]]
     path_ffmpeg = [defaultsettings["MythArchiveFfmpegCmd"], os.path.split(defaultsettings["MythArchiveFfmpegCmd"])[1]]
     path_dvdauthor = [defaultsettings["MythArchiveDvdauthorCmd"], os.path.split(defaultsettings["MythArchiveDvdauthorCmd"])[1]]
-    path_mkisofs = [defaultsettings["MythArchiveMkisofsCmd"], os.path.split(defaultsettings["MythArchiveMkisofsCmd"])[1]]
+    path_genisoimage = [defaultsettings["MythArchiveMkisofsCmd"], os.path.split(defaultsettings["MythArchiveMkisofsCmd"])[1]]
     path_growisofs = [defaultsettings["MythArchiveGrowisofsCmd"], os.path.split(defaultsettings["MythArchiveGrowisofsCmd"])[1]]
     path_tcrequant = [defaultsettings["MythArchiveTcrequantCmd"], os.path.split(defaultsettings["MythArchiveTcrequantCmd"])[1]]
     path_jpeg2yuv = [defaultsettings["MythArchiveJpeg2yuvCmd"], os.path.split(defaultsettings["MythArchiveJpeg2yuvCmd"])[1]]
Index: mythplugins-0.22+fixes20100105/mytharchive/mytharchive/archivesettings.cpp
===================================================================
--- mythplugins-0.22+fixes20100105.orig/mytharchive/mytharchive/archivesettings.cpp	2010-01-18 09:41:22.000000000 +0100
+++ mythplugins-0.22+fixes20100105/mytharchive/mytharchive/archivesettings.cpp	2010-01-18 09:42:15.000000000 +0100
@@ -264,9 +264,9 @@
 static HostLineEdit *MythArchiveMkisofsCmd()
 {
     HostLineEdit *gc = new HostLineEdit("MythArchiveMkisofsCmd");
-    gc->setLabel(QObject::tr("mkisofs command"));
-    gc->setValue("mkisofs");
-    gc->setHelpText(QObject::tr("Command to run mkisofs. (Used to create ISO images)")); 
+    gc->setLabel(QObject::tr("genisoimage command"));
+    gc->setValue("genisoimage");
+    gc->setHelpText(QObject::tr("Command to run genisoimage. (Used to create ISO images)"));
     return gc;
 };
 
