diff -Naur icewm-1.2.21pre1.orig/src/wmapp.cc icewm-1.2.21pre1.patched/src/wmapp.cc
--- icewm-1.2.21pre1.orig/src/wmapp.cc	2006-08-16 14:18:30.000000000 +0000
+++ icewm-1.2.21pre1.patched/src/wmapp.cc	2006-09-04 20:35:52.000000000 +0000
@@ -847,6 +847,14 @@
 
 #ifdef DEBUG
 void dumpZorder(const char *oper, YFrameWindow *w, YFrameWindow *a) {
+    if (oper==null)
+        return;
+    if (w==NULL)
+        return;
+    if (a==null)
+        return;
+    if (manager==NULL)
+        return;
     YFrameWindow *p = manager->top(w->getActiveLayer());
     msg("---- %s ", oper);
     while (p) {
