--- SDL-1.2.7+1.2.8cvs20041007.orig/configure.in	2004-10-06 11:51:12.000000000 -0230
+++ SDL-1.2.7+1.2.8cvs20041007/configure.in	2004-10-06 12:04:57.000000000 -0230
@@ -450,9 +450,9 @@
     if test x$enable_audio = xyes -a x$enable_nas = xyes; then
         AC_MSG_CHECKING(for NAS audio support)
         have_nas=no
-        if test -r /usr/X11R6/include/audio/audiolib.h; then
+        if test -r /usr/include/audio/audiolib.h; then
             have_nas=yes
-            CFLAGS="$CFLAGS -DNAS_SUPPORT -I/usr/X11R6/include/"
+            CFLAGS="$CFLAGS -DNAS_SUPPORT"
             SYSTEM_LIBS="$SYSTEM_LIBS -laudio -lXt"
             AUDIO_SUBDIRS="$AUDIO_SUBDIRS nas"
             AUDIO_DRIVERS="$AUDIO_DRIVERS nas/libaudio_nas.la"
