--- SDL-1.2.7+1.2.8cvs20041007.orig/src/video/x11/SDL_x11modes.c	2004-01-04 13:19:27.000000000 -0330
+++ SDL-1.2.7+1.2.8cvs20041007/src/video/x11/SDL_x11modes.c	2004-12-15 12:59:48.000000000 -0330
@@ -323,7 +323,7 @@
             fclose(metro_fp);
         }
     }
-#if defined(__alpha__) || defined(__sparc64__) || defined(__powerpc__)
+#if defined(__alpha__) || defined(__sparc64__)
     /* The alpha, sparc64 and PPC XFree86 servers are also buggy */
     buggy_X11 = 1;
 #endif
