Index: alsa-plugins-1.0.18/pulse/ctl_pulse.c
===================================================================
--- alsa-plugins-1.0.18.orig/pulse/ctl_pulse.c	2009-02-17 16:23:05.000000000 -0500
+++ alsa-plugins-1.0.18/pulse/ctl_pulse.c	2009-02-17 16:23:20.000000000 -0500
@@ -229,6 +229,11 @@
 					 const snd_ctl_elem_id_t * id)
 {
 	const char *name;
+	unsigned int numid;
+
+	numid = snd_ctl_elem_id_get_numid(id);
+	if (numid > 0 && numid <= 4)
+		return numid - 1;
 
 	name = snd_ctl_elem_id_get_name(id);
 
Index: alsa-plugins-1.0.18/oss/ctl_oss.c
===================================================================
--- alsa-plugins-1.0.18.orig/oss/ctl_oss.c	2009-02-17 16:23:45.000000000 -0500
+++ alsa-plugins-1.0.18/oss/ctl_oss.c	2009-02-17 16:24:07.000000000 -0500
@@ -151,7 +151,21 @@
 {
 	snd_ctl_oss_t *oss = ext->private_data;
 	const char *name;
-	unsigned int i, key;
+	unsigned int i, key, numid;
+
+	numid = snd_ctl_elem_id_get_numid(id);
+	if (numid > 0) {
+		numid--;
+		if (numid < oss->num_vol_ctls)
+			return oss->vol_ctl[numid];
+		numid -= oss->num_vol_ctls;
+		if (oss->exclusive_input) {
+			if (!numid)
+				return OSS_KEY_CAPTURE_MUX;
+		} else if (numid < oss->num_rec_items)
+			return oss->rec_item[numid] |
+				OSS_KEY_CAPTURE_FLAG;
+	}
 
 	name = snd_ctl_elem_id_get_name(id);
 	if (! strcmp(name, "Capture Source")) {
