diff -Nur asterisk-1.2.9.1/channels/Makefile asterisk-1.2.9.1-modified/channels/Makefile
--- asterisk-1.2.9.1/channels/Makefile	2007-01-16 16:30:52.000000000 +0200
+++ asterisk-1.2.9.1-modified/channels/Makefile	2007-01-16 16:30:36.000000000 +0200
@@ -14,6 +14,7 @@
 # the GNU General Public License
 #
 
+CFLAGS += -I/usr/src/linux-headers-`uname -r`/include
 CHANNEL_LIBS=chan_sip.so chan_agent.so chan_mgcp.so chan_iax2.so chan_local.so chan_skinny.so chan_features.so
 
 ifneq (${OSARCH},CYGWIN)
