diff -Nur asterisk-addons-1.2.3/format_mp3/Makefile asterisk-addons-1.2.3-mod/format_mp3/Makefile
--- asterisk-addons-1.2.3/format_mp3/Makefile	2007-01-16 18:19:27.000000000 +0200
+++ asterisk-addons-1.2.3-mod/format_mp3/Makefile	2007-01-16 18:20:09.000000000 +0200
@@ -3,7 +3,7 @@
 MP3OBJS=common.o dct64_i386.o decode_ntom.o layer3.o tabinit.o interface.o format_mp3.o
 
 CFLAGS=-pipe -fPIC -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations $(DEBUG) $(INCLUDE) -D_REENTRANT -D_GNU_SOURCE #-DMAKE_VALGRIND_HAPPY
-
+CFLAGS+=-I../../asterisk/include
 OSARCH=$(shell uname -s)
 
 ifeq (${OSARCH},Linux)
