--- linux/drivers/media/video/saa7127.c.orig	2006-05-07 15:34:59.000000000 +0200
+++ linux/drivers/media/video/saa7127.c	2006-05-07 15:35:12.000000000 +0200
@@ -141,6 +141,7 @@
 static const struct i2c_reg_value saa7129_init_config_extra[] = {
 	{ SAA7127_REG_OUTPUT_PORT_CONTROL, 		0x38 },
 	{ SAA7127_REG_VTRIG, 				0xfa },
+	{ 0, 0 }
 };
 
 static const struct i2c_reg_value saa7127_init_config_common[] = {
