diff -ru icewm-1.2.20/Makefile icewm-1.2.20-alpha/Makefile
--- icewm-1.2.20/Makefile	2005-01-09 16:48:54.000000000 -0500
+++ icewm-1.2.20-alpha/Makefile	2006-07-20 15:15:37.138725961 -0400
@@ -11,7 +11,7 @@
 VERSION = 1.2.20
 
 PREFIX = /usr
-BINDIR = /usr/X11R6/bin
+BINDIR = /usr/bin
 LIBDIR = /usr/share/icewm
 CFGDIR = /etc/icewm
 LOCDIR = /usr/share/locale
@@ -24,7 +24,7 @@
 INSTALLBIN = ${INSTALL}
 INSTALLLIB = ${INSTALL} -m 644
 INSTALLETC = ${INSTALL} -m 644
-MKFONTDIR  = /usr/X11R6/bin/mkfontdir
+MKFONTDIR  = /usr/bin/mkfontdir
 
 DESTDIR     =
 
Only in icewm-1.2.20/lib: keys
Only in icewm-1.2.20/lib: menu
Only in icewm-1.2.20/lib: programs
Only in icewm-1.2.20/lib: toolbar
Only in icewm-1.2.20/lib: winoptions
Only in icewm-1.2.20-alpha/po: be.mo
Only in icewm-1.2.20-alpha/po: bg.mo
Only in icewm-1.2.20-alpha/po: ca.mo
Only in icewm-1.2.20-alpha/po: cs.mo
Only in icewm-1.2.20-alpha/po: de.mo
Only in icewm-1.2.20-alpha/po: el.mo
Only in icewm-1.2.20-alpha/po: en.mo
Only in icewm-1.2.20-alpha/po: es.mo
Only in icewm-1.2.20-alpha/po: fi.mo
Only in icewm-1.2.20-alpha/po: fr.mo
Only in icewm-1.2.20-alpha/po: hr.mo
Only in icewm-1.2.20-alpha/po: hu.mo
Only in icewm-1.2.20-alpha/po: it.mo
Only in icewm-1.2.20-alpha/po: ja.mo
Only in icewm-1.2.20-alpha/po: ko.mo
Only in icewm-1.2.20-alpha/po: lt.mo
Only in icewm-1.2.20-alpha/po: nl.mo
Only in icewm-1.2.20-alpha/po: no.mo
Only in icewm-1.2.20-alpha/po: pl.mo
Only in icewm-1.2.20-alpha/po: pt_BR.mo
Only in icewm-1.2.20-alpha/po: ro.mo
Only in icewm-1.2.20-alpha/po: ru.mo
Only in icewm-1.2.20-alpha/po: sk.mo
Only in icewm-1.2.20-alpha/po: sl.mo
Only in icewm-1.2.20-alpha/po: sv.mo
Only in icewm-1.2.20-alpha/po: tr.mo
Only in icewm-1.2.20-alpha/po: uk.mo
Only in icewm-1.2.20-alpha/po: zh_CN.mo
Only in icewm-1.2.20-alpha/po: zh_TW.Big5.mo
diff -ru icewm-1.2.20/src/Makefile icewm-1.2.20-alpha/src/Makefile
--- icewm-1.2.20/src/Makefile	2005-01-09 16:48:54.000000000 -0500
+++ icewm-1.2.20-alpha/src/Makefile	2006-07-20 15:15:37.525660173 -0400
@@ -1,7 +1,7 @@
 PACKAGE = icewm
 VERSION = 1.2.20
-HOSTOS  = Linux 2.6.9-1.681_FC3
-HOSTCPU = i686
+HOSTOS  = Linux 2.6.16.16
+HOSTCPU = x86_64
 
 LIBDIR = /usr/share/icewm
 CFGDIR = /etc/icewm
@@ -34,12 +34,12 @@
                 -DICEHELPIDX='"$(DOCDIR)/icewm-$(VERSION)/icewm.html"'
 
 CXXFLAGS =      -fpermissive -Wall -Wpointer-arith -Wconversion -Wwrite-strings -Winline -Woverloaded-virtual -W -fno-exceptions -fno-rtti -g -O2 $(DEBUG) $(DEFS) \
-	         -I/usr/X11R6/include -I/usr/X11R6/include -I/usr/include/freetype2 -I/usr/include/freetype2/config -I/usr/X11R6/include -I/usr/X11R6/include  # `fc-config --cflags`
+	         -I/usr/include -I/usr/include/freetype2   # `fc-config --cflags`
 LFLAGS =	
 LIBS =          
 
-CORE_LIBS =     -lXinerama  -lSM -lICE  -lX11  -L/usr/X11R6/lib  -lXrandr -lXrender -lXext -L/usr/X11R6/lib -lXft -lX11 -lfreetype -lfontconfig -L/usr/X11R6/lib -lXrender # `fc-config --libs`
-IMAGE_LIBS =    -L/usr/lib -lImlib -ljpeg -ltiff -lungif -lpng -lz -lm -lXext -L/usr/X11R6/lib -lSM -lICE -lXext -lX11
+CORE_LIBS =     -lXinerama  -lSM -lICE  -lX11  -L/usr/lib64  -lXrandr -lXrender -lXext -L/usr/lib64 -lXft -lX11 -lfreetype -lz -lfontconfig -lXrender -lX11 # `fc-config --libs`
+IMAGE_LIBS =    -L/usr/lib64 -lImlib -ljpeg -ltiff -lgif -lpng -lz -lm -lXext -L/usr/lib64 -lSM -lICE -lXext -lX11
 AUDIO_LIBS =    
 GNOME1_LIBS =    
 GNOME2_LIBS =    
Only in icewm-1.2.20/src: config.h
diff -ru icewm-1.2.20/src/wmcontainer.cc icewm-1.2.20-alpha/src/wmcontainer.cc
--- icewm-1.2.20/src/wmcontainer.cc	2005-01-09 16:48:23.000000000 -0500
+++ icewm-1.2.20-alpha/src/wmcontainer.cc	2006-07-20 15:11:57.749035101 -0400
@@ -14,8 +14,9 @@
 
 #include <stdio.h>
 
-YClientContainer::YClientContainer(YWindow *parent, YFrameWindow *frame)
-:YWindow(parent)
+YClientContainer::YClientContainer(YWindow *parent, YFrameWindow *frame,
+                                   Visual *visual)
+    :YWindow(parent, 0, visual)
 {
     fFrame = frame;
     fHaveGrab = false;
diff -ru icewm-1.2.20/src/wmcontainer.h icewm-1.2.20-alpha/src/wmcontainer.h
--- icewm-1.2.20/src/wmcontainer.h	2005-01-09 16:48:23.000000000 -0500
+++ icewm-1.2.20-alpha/src/wmcontainer.h	2006-07-20 15:11:57.749035101 -0400
@@ -7,7 +7,7 @@
 
 class YClientContainer: public YWindow {
 public:
-    YClientContainer(YWindow *parent, YFrameWindow *frame);
+    YClientContainer(YWindow *parent, YFrameWindow *frame, Visual *visual);
     virtual ~YClientContainer();
 
     virtual void handleButton(const XButtonEvent &button);
diff -ru icewm-1.2.20/src/wmframe.cc icewm-1.2.20-alpha/src/wmframe.cc
--- icewm-1.2.20/src/wmframe.cc	2005-01-09 16:48:24.000000000 -0500
+++ icewm-1.2.20-alpha/src/wmframe.cc	2006-07-20 15:11:57.751034761 -0400
@@ -48,7 +48,8 @@
     return false;
 }
 
-YFrameWindow::YFrameWindow(YWindow *parent): YWindow(parent) {
+YFrameWindow::YFrameWindow(YWindow *parent, Visual *visual)
+    : YWindow(parent, 0, visual) {
     if (activeBorderBg == 0)
         activeBorderBg = new YColor(clrActiveBorder);
     if (inactiveBorderBg == 0)
@@ -126,7 +127,7 @@
 
     createPointerWindows();
 
-    fClientContainer = new YClientContainer(this, this);
+    fClientContainer = new YClientContainer(this, this, visual);
     fClientContainer->show();
 
     fTitleBar = new YFrameTitleBar(this, this);
diff -ru icewm-1.2.20/src/wmframe.h icewm-1.2.20-alpha/src/wmframe.h
--- icewm-1.2.20/src/wmframe.h	2005-01-09 16:48:24.000000000 -0500
+++ icewm-1.2.20-alpha/src/wmframe.h	2006-07-20 15:11:57.752034591 -0400
@@ -20,7 +20,7 @@
 
 class YFrameWindow: public YWindow, public YActionListener, public YTimerListener, public YPopDownListener, public YMsgBoxListener, public ClientData {
 public:
-    YFrameWindow(YWindow *parent);
+    YFrameWindow(YWindow *parent, Visual *visual = 0);
     virtual ~YFrameWindow();
 
     void doManage(YFrameClient *client);
diff -ru icewm-1.2.20/src/wmmgr.cc icewm-1.2.20-alpha/src/wmmgr.cc
--- icewm-1.2.20/src/wmmgr.cc	2005-01-09 16:48:23.000000000 -0500
+++ icewm-1.2.20-alpha/src/wmmgr.cc	2006-07-20 15:11:57.753034421 -0400
@@ -1355,7 +1355,7 @@
 
     manager->updateFullscreenLayerEnable(false);
 
-    frame = new YFrameWindow(0);
+    frame = new YFrameWindow(0, client->visual());
     if (frame == 0) {
         delete client;
         goto end;
Only in icewm-1.2.20-alpha/src: wmmgr.cc.orig
diff -ru icewm-1.2.20/src/ywindow.cc icewm-1.2.20-alpha/src/ywindow.cc
--- icewm-1.2.20/src/ywindow.cc	2005-01-09 16:48:24.000000000 -0500
+++ icewm-1.2.20-alpha/src/ywindow.cc	2006-07-20 15:13:41.092458507 -0400
@@ -18,6 +18,8 @@
 #include "ytimer.h"
 #include "ypopup.h"
 
+#include <X11/extensions/Xrender.h>
+
 /******************************************************************************/
 /******************************************************************************/
 
@@ -100,13 +102,13 @@
 /******************************************************************************/
 /******************************************************************************/
 
-YWindow::YWindow(YWindow *parent, Window win):
+YWindow::YWindow(YWindow *parent, Window win, Visual *visual):
     fParentWindow(parent),
     fNextWindow(0), fPrevWindow(0), fFirstWindow(0), fLastWindow(0),
     fFocusedWindow(0),
 
     fHandle(win), flags(0), fStyle(0), fX(0), fY(0), fWidth(1), fHeight(1),
-    fPointer(), unmapCount(0), fGraphics(0),
+    fPointer(), unmapCount(0), fGraphics(0), fVisual(visual),
     fEventMask(KeyPressMask|KeyReleaseMask|FocusChangeMask|
                LeaveWindowMask|EnterWindowMask),
     fWinGravity(NorthWestGravity), fBitGravity(ForgetGravity),
@@ -286,7 +288,7 @@
                                 0,
                                 CopyFromParent,
                                 (fStyle & wsInputOnly) ? InputOnly : InputOutput,
-                                CopyFromParent,
+                                (fVisual) ? fVisual : CopyFromParent,
                                 attrmask,
                                 &attributes);
 
@@ -296,6 +298,9 @@
 
         if ((flags & wfVisible) && !(flags & wfNullSize))
             XMapWindow(xapp->display(), fHandle);
+
+        // Zero visual so it is not used for system menu
+        fVisual = 0;
     } else {
         XWindowAttributes attributes;
 
@@ -337,6 +342,22 @@
         }
 
         XSelectInput(xapp->display(), fHandle, fEventMask);
+
+        // Get the visual if we have a window with alpha transparency
+        XWindowAttributes attr;
+        Status err = XGetWindowAttributes(xapp->display(), fHandle, &attr);
+        bool ok = (BadDrawable != err) && (BadWindow != err) && attr.visual;
+
+        XRenderPictFormat *format = NULL;
+        if (ok)
+            format = XRenderFindVisualFormat(xapp->display(), attr.visual);
+
+        if (format && format->type == PictTypeDirect &&
+            format->direct.alphaMask)
+        {
+            //printf("YWindow(0x%X) adopted alpha window\n", this);
+            fVisual = attr.visual;
+        }
     }
     XSaveContext(xapp->display(), fHandle, windowContext, (XPointer)this);
     flags |= wfCreated;
Only in icewm-1.2.20-alpha/src: ywindow.cc.orig
Only in icewm-1.2.20-alpha/src: ywindow.cc.rej
diff -ru icewm-1.2.20/src/ywindow.h icewm-1.2.20-alpha/src/ywindow.h
--- icewm-1.2.20/src/ywindow.h	2005-01-09 16:48:24.000000000 -0500
+++ icewm-1.2.20-alpha/src/ywindow.h	2006-07-20 15:11:57.788028468 -0400
@@ -24,7 +24,7 @@
 
 class YWindow {
 public:
-    YWindow(YWindow *aParent = 0, Window win = 0);
+    YWindow(YWindow *aParent = 0, Window win = 0, Visual *visual = 0);
     virtual ~YWindow();
 
     void setStyle(unsigned long aStyle);
@@ -202,6 +202,8 @@
     bool hasPopup();
     void setDoubleBuffer(bool doubleBuffer);
 
+    Visual *visual() { return fVisual; }
+
 private:
     typedef enum {
         wfVisible   = 1 << 0,
@@ -235,6 +237,7 @@
     YCursor fPointer;
     int unmapCount;
     Graphics *fGraphics;
+    Visual *fVisual;
     long fEventMask;
     int fWinGravity, fBitGravity;
 
Only in icewm-1.2.20-alpha/src: ywindow.h.orig
