diff -r -u AMP.orig/amp_conf/var/www/html/recordings/index.php AMP/amp_conf/var/www/html/recordings/index.php
--- AMP.orig/amp_conf/var/www/html/recordings/index.php	2005-08-20 01:02:20.000000000 +0300
+++ AMP/amp_conf/var/www/html/recordings/index.php	2006-02-10 14:45:44.000000000 +0200
@@ -1,4 +1,31 @@
 <?php
+// *********** autologin for pluto admin ******************
+session_name('Pluto-admin');
+session_start();
+
+if(isset($_SESSION['Extension'])){
+
+	unset($_SESSION['ari_user']);
+	$_POST['username']=$_SESSION['Extension'];
+	$_POST['password']=read_pass_from_file($_SESSION['Extension'],'/etc/asterisk/voicemail.conf');
+}
+
+function read_pass_from_file($username,$file){
+	$lines = file($file);
+	foreach ($lines as $key => $line) {
+
+		unset($value);
+		list($var,$value) = split('=>',$line);
+		$var = trim($var);
+
+		if ($var==$username && $value) {
+			$buf = split(',',$value);
+			return $buf[0];
+		}
+	}
+	return false;
+}
+// *********** end autologin ******************
 
 /**
  * @file
Only in AMP/amp_conf/var/www/html/recordings: index.php.orig
diff -r -u AMP.orig/amp_conf/var/www/html/recordings/theme/page.tpl.php AMP/amp_conf/var/www/html/recordings/theme/page.tpl.php
--- AMP.orig/amp_conf/var/www/html/recordings/theme/page.tpl.php	2005-08-20 01:02:23.000000000 +0300
+++ AMP/amp_conf/var/www/html/recordings/theme/page.tpl.php	2006-02-10 14:49:19.000000000 +0200
@@ -36,7 +36,7 @@
                 <p><small><small><a href='<?=$_SERVER['PHP_SELF']?>?s=voicemail'>Voicemail</a></small></small></p>
               <?php } ?>
                 <p><small><small><a href='<?=$_SERVER['PHP_SELF']?>?s=callmonitor'>Call Monitor</a></small></small></p>
-                <?php if ($logout !='') { ?>
+                <?php if ($logout !='' && !isset($_SESSION['Extension'])) { ?>
                   <br>
                   <p><small><small><a href='<?=$_SERVER['PHP_SELF']?>?logout=1'>Logout</a></small></small></p>
                 <?php } ?>
Only in AMP/amp_conf/var/www/html/recordings/theme: page.tpl.php.orig
