diff -Naur dbus-0.61.orig/mono/AssemblyInfo.cs.in dbus-0.61/mono/AssemblyInfo.cs.in
--- dbus-0.61.orig/mono/AssemblyInfo.cs.in	2006-03-01 19:58:32.000000000 +0100
+++ dbus-0.61/mono/AssemblyInfo.cs.in	2006-03-01 19:59:00.000000000 +0100
@@ -1,6 +1,6 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
  
-[assembly:AssemblyVersion("@VERSION@")]
+[assembly:AssemblyVersion("0.60")]
 [assembly:AssemblyDelaySign(false)]
 [assembly:AssemblyKeyFile("@srcdir@/dbus-sharp.snk")]
