Index: qt/qdbusintegrator.cpp
===================================================================
RCS file: /cvs/dbus/dbus/qt/qdbusintegrator.cpp,v
retrieving revision 1.3
diff -u -r1.3 qdbusintegrator.cpp
--- qt/qdbusintegrator.cpp	20 Feb 2006 13:23:11 -0000	1.3
+++ qt/qdbusintegrator.cpp	28 Feb 2006 09:32:47 -0000
@@ -39,6 +39,15 @@
 
 int QDBusConnectionPrivate::messageMetaType = 0;
 
+#if __BYTE_ORDER != __LITTLE_ENDIAN
+    union integer
+    {
+        short s;
+        unsigned short us;
+        unsigned char uc;
+    };
+#endif
+
 struct QDBusPendingCall
 {
     QPointer<QObject> receiver;
@@ -622,12 +631,6 @@
     params.reserve(metaTypes.count());
 
 #if __BYTE_ORDER != __LITTLE_ENDIAN
-    union integer
-    {
-        short s;
-        unsigned short us;
-        unsigned char uc;
-    }
     QVarLengthArray<integer, 4> auxParameters;
 #endif
     // let's create the parameter list
