diff -ruN hal-0.5.6+cvs20060217-old/hal.conf.in hal-0.5.6+cvs20060217/hal.conf.in
--- hal-0.5.6+cvs20060217-old/hal.conf.in	2006-02-17 21:15:40.000000000 +0100
+++ hal-0.5.6+cvs20060217/hal.conf.in	2006-02-20 23:29:14.000000000 +0100
@@ -37,16 +37,20 @@
   <policy at_console="true">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <!-- 
     <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+    -->
   </policy>
 
   <!-- You can change this to a more suitable user, or make per-group -->
-  <policy user="0">
+  <policy group="powerdev">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <!-- 
     <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+    -->
   </policy>
 
 </busconfig>
