diff -ruN hal-0.5.7-old/hald/linux2/addons/addon-keyboard.c hal-0.5.7/hald/linux2/addons/addon-keyboard.c
--- hal-0.5.7-old/hald/linux2/addons/addon-keyboard.c	2006-02-22 22:10:06.000000000 +0100
+++ hal-0.5.7/hald/linux2/addons/addon-keyboard.c	2006-03-05 22:02:49.000000000 +0100
@@ -1,5 +1,5 @@
 /***************************************************************************
- * CVSID: $Id: addon-keyboard.c,v 1.4 2006/02/22 21:10:06 david Exp $
+ * CVSID: $Id: addon-keyboard.c,v 1.5 2006/03/03 17:33:26 hughsient Exp $
  *
  * addon-keyboard.c : Listen to key events and modify hal device objects
  *
@@ -62,10 +62,10 @@
 	[KEY_CALC] = "calc",
 	[KEY_SETUP] = "setup",
 	[KEY_SLEEP] = "sleep",
-	[KEY_WAKEUP] = "wakeup",
+	[KEY_WAKEUP] = "wake-up",
 	[KEY_FILE] = "file",
-	[KEY_SENDFILE] = "sendfile",
-	[KEY_DELETEFILE] = "deletefile",
+	[KEY_SENDFILE] = "send-file",
+	[KEY_DELETEFILE] = "delete-file",
 	[KEY_XFER] = "xfer",
 	[KEY_PROG1] = "prog1",
 	[KEY_PROG2] = "prog2",
@@ -73,19 +73,19 @@
 	[KEY_MSDOS] = "msdos",
 	[KEY_COFFEE] = "coffee",
 	[KEY_DIRECTION] = "direction",
-	[KEY_CYCLEWINDOWS] = "cyclewindows",
+	[KEY_CYCLEWINDOWS] = "cycle-windows",
 	[KEY_MAIL] = "mail",
 	[KEY_BOOKMARKS] = "bookmarks",
 	[KEY_COMPUTER] = "computer",
 	[KEY_BACK] = "back",
 	[KEY_FORWARD] = "forward",
-	[KEY_CLOSECD] = "closecd",
-	[KEY_EJECTCD] = "ejectcd",
-	[KEY_EJECTCLOSECD] = "ejectclosecd",
-	[KEY_NEXTSONG] = "nextsong",
-	[KEY_PLAYPAUSE] = "playpause",
-	[KEY_PREVIOUSSONG] = "previoussong",
-	[KEY_STOPCD] = "stopcd",
+	[KEY_CLOSECD] = "close-cd",
+	[KEY_EJECTCD] = "eject-cd",
+	[KEY_EJECTCLOSECD] = "eject-close-cd",
+	[KEY_NEXTSONG] = "next-song",
+	[KEY_PLAYPAUSE] = "play-pause",
+	[KEY_PREVIOUSSONG] = "previous-song",
+	[KEY_STOPCD] = "stop-cd",
 	[KEY_RECORD] = "record",
 	[KEY_REWIND] = "rewind",
 	[KEY_PHONE] = "phone",
@@ -96,10 +96,10 @@
 	[KEY_EXIT] = "exit",
 	[KEY_MOVE] = "move",
 	[KEY_EDIT] = "edit",
-	[KEY_SCROLLUP] = "scrollup",
-	[KEY_SCROLLDOWN] = "scrolldown",
-	[KEY_KPLEFTPAREN] = "kpleftparen",
-	[KEY_KPRIGHTPAREN] = "kprightparen",
+	[KEY_SCROLLUP] = "scroll-up",
+	[KEY_SCROLLDOWN] = "scroll-down",
+	[KEY_KPLEFTPAREN] = "kp-left-paren",
+	[KEY_KPRIGHTPAREN] = "kp-right-paren",
 	[KEY_F13] = "f13",
 	[KEY_F14] = "f14",
 	[KEY_F15] = "f15",
@@ -112,15 +112,15 @@
 	[KEY_F22] = "f22",
 	[KEY_F23] = "f23",
 	[KEY_F24] = "f24",
-	[KEY_PLAYCD] = "playcd",
-	[KEY_PAUSECD] = "pausecd",
+	[KEY_PLAYCD] = "play-cd",
+	[KEY_PAUSECD] = "pause-cd",
 	[KEY_PROG3] = "prog3",
 	[KEY_PROG4] = "prog4",
 	[KEY_SUSPEND] = "hibernate",
 	[KEY_CLOSE] = "close",
 	[KEY_PLAY] = "play",
-	[KEY_FASTFORWARD] = "fastforward",
-	[KEY_BASSBOOST] = "bassboost",
+	[KEY_FASTFORWARD] = "fast-forward",
+	[KEY_BASSBOOST] = "bass-boost",
 	[KEY_PRINT] = "print",
 	[KEY_HP] = "hp",
 	[KEY_CAMERA] = "camera",
@@ -133,12 +133,31 @@
 	[KEY_FINANCE] = "finance",
 	[KEY_SPORT] = "sport",
 	[KEY_SHOP] = "shop",
-	[KEY_ALTERASE] = "alterase",
+	[KEY_ALTERASE] = "alt-erase",
 	[KEY_CANCEL] = "cancel",
-	[KEY_BRIGHTNESSDOWN] = "brightnessdown",
-	[KEY_BRIGHTNESSUP] = "brightnessup",
+	[KEY_BRIGHTNESSDOWN] = "brightness-down",
+	[KEY_BRIGHTNESSUP] = "brightness-up",
 	[KEY_MEDIA] = "media",
-	[KEY_POWER] = "power"
+	[KEY_POWER] = "power",
+	[KEY_MUTE] = "mute",
+	[KEY_VOLUMEDOWN] = "volume-down",
+	[KEY_VOLUMEUP] = "volume-up",
+#ifndef KEY_SWITCHVIDEOMODE
+#define KEY_SWITCHVIDEOMODE	227
+#endif
+	[KEY_SWITCHVIDEOMODE] = "switch-videomode",
+#ifndef KEY_KBDILLUMTOGGLE
+#define KEY_KBDILLUMTOGGLE	228
+#endif
+	[KEY_KBDILLUMTOGGLE] = "kbd-illum-toggle",
+#ifndef KEY_KBDILLUMDOWN
+#define KEY_KBDILLUMDOWN	229
+#endif
+	[KEY_KBDILLUMDOWN] = "kbd-illum-down",
+#ifndef KEY_KBDILLUMUP
+#define KEY_KBDILLUMUP		230
+#endif
+	[KEY_KBDILLUMUP] = "kbd-illum-up"
 };
 
 static void
