diff -ruN hal-0.5.7-old/tools/hal-system-power-suspend hal-0.5.7/tools/hal-system-power-suspend
--- hal-0.5.7-old/tools/hal-system-power-suspend	2006-05-01 22:22:24.000000000 +0200
+++ hal-0.5.7/tools/hal-system-power-suspend	2006-05-01 22:23:30.000000000 +0200
@@ -50,6 +50,10 @@
 	if [ -x "/usr/bin/powersave" ] ; then
 	    $POWERSAVED_SUSPEND2RAM
 	    RET=$?
+	elif [ -x "/usr/sbin/hibernate" ] ; then
+		# Use hibernate configured for suspend-to-ram
+		/usr/sbin/hibernate -F/etc/hibernate/ram.conf
+		RET=$?
 	elif [ -x "/usr/sbin/pmi" ] ; then
 	    /usr/sbin/pmi action suspend force
 	    RET=$?
