diff -ruN hal-0.5.7-old/hald/linux2/classdev.c hal-0.5.7/hald/linux2/classdev.c
--- hal-0.5.7-old/hald/linux2/classdev.c	2006-01-28 23:51:18.000000000 +0100
+++ hal-0.5.7/hald/linux2/classdev.c	2006-05-01 22:42:45.000000000 +0200
@@ -1,5 +1,5 @@
 /***************************************************************************
- * CVSID: $Id: classdev.c,v 1.42 2006/01/28 22:51:18 david Exp $
+ * CVSID: $Id: classdev.c,v 1.43 2006/03/03 19:19:53 dkukawka Exp $
  *
  * classdev.c : Handling of functional kernel devices
  *
@@ -927,7 +927,12 @@
 static const gchar *
 serial_get_prober (HalDevice *d)
 {
-	return "hald-probe-serial";
+	/* FIXME TODO: check if there is an other way, to call the porber only
+		 on ttyS* devices, than check the name of the device file */
+	if (!strncmp(hal_device_property_get_string (d, "linux.device_file"), "/dev/ttyS", 9))
+		return "hald-probe-serial";
+	else 
+		return NULL;
 }
 
 static gboolean
