Index: vdr-plugin-softhddevice-0.6.1rc1.git20131220.2124/softhddev.c
===================================================================
--- vdr-plugin-softhddevice-0.6.1rc1.git20131220.2124.orig/softhddev.c	2013-12-20 22:24:16.000000000 +0100
+++ vdr-plugin-softhddevice-0.6.1rc1.git20131220.2124/softhddev.c	2014-01-01 13:05:08.382045875 +0100
@@ -33,6 +33,7 @@
 
 #include <stdio.h>
 #include <stdint.h>
+#include <string.h>
 #include <inttypes.h>
 #include <unistd.h>
 
@@ -41,6 +42,7 @@
 #define _N(str) str			///< gettext_noop shortcut
 
 #include <libavcodec/avcodec.h>
+#include <libavutil/mem.h>
 // support old ffmpeg versions <1.0
 #if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(55,18,102)
 #define AVCodecID CodecID
