diff --git a/video.c b/video.c
index a5b3a5a..30acf5e 100644
--- a/video.c
+++ b/video.c
@@ -9736,6 +9736,7 @@ static void VideoEvent(void)
     const char *keynam;
     char buf[64];
     char letter[64];
+    int  letter_len;
     uint32_t values[1];
 
     VideoThreadLock();
@@ -9775,7 +9776,10 @@ static void VideoEvent(void)
 	    break;
 	case KeyPress:
 	    VideoThreadLock();
-	    XLookupString(&event.xkey, letter, sizeof(letter), &keysym, NULL);
+	    letter_len = XLookupString(&event.xkey, letter, sizeof(letter) - 1, &keysym, NULL);
+	    if (letter_len < 0)
+	        letter_len = 0;
+	    letter[letter_len] = 0;
 	    VideoThreadUnlock();
 	    if (keysym == NoSymbol) {
 		Warning(_("video/event: No symbol for %d\n"),
