diff -ruN mplayer/mplayer.c mplayer.verbose/mplayer.c
--- mplayer/mplayer.c	2007-10-30 23:44:44.000000000 +0200
+++ mplayer.verbose/mplayer.c	2007-10-30 23:44:36.000000000 +0200
@@ -3684,11 +3684,20 @@
 #endif
 
 if(use_gui || mpctx->playtree_iter != NULL || player_idle_mode){
+  if (filename) {
+	  printf("Playback finished: %s.\n", filename);
+	  fflush(stdout);
+  }
+
   if (!mpctx->playtree_iter) filename = NULL;
   mpctx->eof = 0;
   goto play_next_file;
 }
 
+if (filename) {
+	printf("Playback finished: %s.\n", filename);
+        fflush(stdout);
+}
 
 exit_player_with_rc(MSGTR_Exit_eof, 0);
 
