diff -ruN mplayer/mplayer.c mplayer.patched/mplayer.c
--- mplayer/mplayer.c	2007-11-03 21:17:13.000000000 +0200
+++ mplayer.patched/mplayer.c	2007-11-03 21:16:57.000000000 +0200
@@ -3030,6 +3030,9 @@
         filename = play_tree_iter_get_file(mpctx->playtree_iter, 1);
     }
 }
+
+if (filename) load_per_file_config (mconfig, filename);
+
 //---------------------------------------------------------------------------
 
     if(filename)
