diff -ruN mplayer/mplayer.c mplayer.patched/mplayer.c
--- mplayer/mplayer.c	2007-11-03 21:17:13.000000000 +0200
+++ mplayer.patched/mplayer.c	2007-11-03 21:16:57.000000000 +0200
@@ -2675,8 +2675,6 @@
   mpctx->global_sub_size = 0;
   { int i; for (i = 0; i < SUB_SOURCES; i++) mpctx->global_sub_indices[i] = -1; }
 
-  if (filename) load_per_file_config (mconfig, filename);
-
 // We must enable getch2 here to be able to interrupt network connection
 // or cache filling
 if(!noconsolecontrols && !slave_mode){
@@ -2773,6 +2771,9 @@
         filename = play_tree_iter_get_file(mpctx->playtree_iter, 1);
     }
 }
+
+if (filename) load_per_file_config (mconfig, filename);
+
 //---------------------------------------------------------------------------
 
     if(filename)
