diff -Naur icewm-1.2.28pre2.orig/VERSION icewm-1.2.28pre2.patched/VERSION
--- icewm-1.2.28pre2.orig/VERSION	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/VERSION	2006-09-05 15:31:47.000000000 +0000
@@ -1,2 +1,2 @@
 PACKAGE=icewm
-VERSION=1.2.28pre2
+VERSION=1.2.28pre2-alphapatch
diff -Naur icewm-1.2.28pre2.orig/src/Makefile.in icewm-1.2.28pre2.patched/src/Makefile.in
--- icewm-1.2.28pre2.orig/src/Makefile.in	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/Makefile.in	2006-09-05 15:33:22.000000000 +0000
@@ -38,7 +38,7 @@
 LFLAGS =	@LDFLAGS@
 LIBS =          @LIBS@
 
-CORE_LIBS =     @CORE_LIBS@ # `fc-config --libs`
+CORE_LIBS =     @CORE_LIBS@ -lXrender # `fc-config --libs`
 IMAGE_LIBS =    @IMAGE_LIBS@
 AUDIO_LIBS =    @AUDIO_LIBS@
 GNOME1_LIBS =    @GNOME1_LIBS@
diff -Naur icewm-1.2.28pre2.orig/src/wmcontainer.cc icewm-1.2.28pre2.patched/src/wmcontainer.cc
--- icewm-1.2.28pre2.orig/src/wmcontainer.cc	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/wmcontainer.cc	2006-09-05 15:30:36.000000000 +0000
@@ -14,8 +14,8 @@
 
 #include <stdio.h>
 
-YClientContainer::YClientContainer(YWindow *parent, YFrameWindow *frame)
-:YWindow(parent)
+YClientContainer::YClientContainer(YWindow *parent, YFrameWindow *frame, Visual *visual)
+    :YWindow(parent, 0, visual)
 {
     fFrame = frame;
     fHaveGrab = false;
diff -Naur icewm-1.2.28pre2.orig/src/wmcontainer.h icewm-1.2.28pre2.patched/src/wmcontainer.h
--- icewm-1.2.28pre2.orig/src/wmcontainer.h	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/wmcontainer.h	2006-09-05 15:30:36.000000000 +0000
@@ -7,7 +7,7 @@
 
 class YClientContainer: public YWindow {
 public:
-    YClientContainer(YWindow *parent, YFrameWindow *frame);
+    YClientContainer(YWindow *parent, YFrameWindow *frame, Visual *visual);
     virtual ~YClientContainer();
 
     virtual void handleButton(const XButtonEvent &button);
diff -Naur icewm-1.2.28pre2.orig/src/wmframe.cc icewm-1.2.28pre2.patched/src/wmframe.cc
--- icewm-1.2.28pre2.orig/src/wmframe.cc	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/wmframe.cc	2006-09-05 15:30:36.000000000 +0000
@@ -53,7 +53,7 @@
     return false;
 }
 
-YFrameWindow::YFrameWindow(YWindow *parent): YWindow(parent) {
+YFrameWindow::YFrameWindow(YWindow *parent, Visual *visual) : YWindow(parent, 0, visual) {
     if (activeBorderBg == 0)
         activeBorderBg = new YColor(clrActiveBorder);
     if (inactiveBorderBg == 0)
@@ -129,7 +129,7 @@
 
     createPointerWindows();
 
-    fClientContainer = new YClientContainer(this, this);
+    fClientContainer = new YClientContainer(this, this, visual);
     fClientContainer->show();
 
     fTitleBar = new YFrameTitleBar(this, this);
diff -Naur icewm-1.2.28pre2.orig/src/wmframe.h icewm-1.2.28pre2.patched/src/wmframe.h
--- icewm-1.2.28pre2.orig/src/wmframe.h	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/wmframe.h	2006-09-05 15:30:36.000000000 +0000
@@ -20,7 +20,7 @@
 
 class YFrameWindow: public YWindow, public YActionListener, public YTimerListener, public YPopDownListener, public YMsgBoxListener, public ClientData {
 public:
-    YFrameWindow(YWindow *parent);
+    YFrameWindow(YWindow *parent, Visual *visual = 0);
     virtual ~YFrameWindow();
 
     void doManage(YFrameClient *client, bool &doActivate);
diff -Naur icewm-1.2.28pre2.orig/src/wmmgr.cc icewm-1.2.28pre2.patched/src/wmmgr.cc
--- icewm-1.2.28pre2.orig/src/wmmgr.cc	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/wmmgr.cc	2006-09-05 15:30:36.000000000 +0000
@@ -1394,7 +1394,7 @@
 
     manager->updateFullscreenLayerEnable(false);
 
-    frame = new YFrameWindow(0);
+    frame = new YFrameWindow(0, client->visual());
     if (frame == 0) {
         delete client;
         goto end;
diff -Naur icewm-1.2.28pre2.orig/src/ywindow.cc icewm-1.2.28pre2.patched/src/ywindow.cc
--- icewm-1.2.28pre2.orig/src/ywindow.cc	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/ywindow.cc	2006-09-05 15:30:36.000000000 +0000
@@ -18,6 +18,8 @@
 #include "ytimer.h"
 #include "ypopup.h"
 
+#include <X11/extensions/Xrender.h>
+
 /******************************************************************************/
 /******************************************************************************/
 
@@ -121,7 +123,7 @@
 /******************************************************************************/
 /******************************************************************************/
 
-YWindow::YWindow(YWindow *parent, Window win):
+YWindow::YWindow(YWindow *parent, Window win, Visual *visual):
     fParentWindow(parent),
     fNextWindow(0), fPrevWindow(0), fFirstWindow(0), fLastWindow(0),
     fFocusedWindow(0),
@@ -129,6 +131,7 @@
     fHandle(win), flags(0), fStyle(0), fX(0), fY(0), fWidth(1), fHeight(1),
     fPointer(), unmapCount(0), 
     fGraphics(0),
+    fVisual(visual),
     fEventMask(KeyPressMask|KeyReleaseMask|FocusChangeMask|
                LeaveWindowMask|EnterWindowMask),
     fWinGravity(NorthWestGravity), fBitGravity(ForgetGravity),
@@ -308,7 +311,7 @@
                                 0,
                                 CopyFromParent,
                                 (fStyle & wsInputOnly) ? InputOnly : InputOutput,
-                                CopyFromParent,
+                                (fVisual) ? fVisual : CopyFromParent,
                                 attrmask,
                                 &attributes);
 
@@ -318,6 +321,9 @@
 
         if ((flags & wfVisible) && !(flags & wfNullSize))
             XMapWindow(xapp->display(), fHandle);
+
+        // Zero visual so it is not used for system menu
+        fVisual = 0;
     } else {
         XWindowAttributes attributes;
 
@@ -359,6 +365,22 @@
         }
 
         XSelectInput(xapp->display(), fHandle, fEventMask);
+
+        // Get the visual if we have a window with alpha transparency
+        XWindowAttributes attr;
+        Status err = XGetWindowAttributes(xapp->display(), fHandle, &attr);
+        bool ok = (BadDrawable != err) && (BadWindow != err) && attr.visual;
+
+        XRenderPictFormat *format = NULL;
+        if (ok)
+            format = XRenderFindVisualFormat(xapp->display(), attr.visual);
+
+        if (format && format->type == PictTypeDirect &&
+            format->direct.alphaMask)
+        {
+            //printf("YWindow(0x%X) adopted alpha window\n", this);
+            fVisual = attr.visual;
+        }
     }
     XSaveContext(xapp->display(), fHandle, windowContext, (XPointer)this);
     flags |= wfCreated;
diff -Naur icewm-1.2.28pre2.orig/src/ywindow.h icewm-1.2.28pre2.patched/src/ywindow.h
--- icewm-1.2.28pre2.orig/src/ywindow.h	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/ywindow.h	2006-09-05 15:30:36.000000000 +0000
@@ -24,7 +24,7 @@
 
 class YWindow {
 public:
-    YWindow(YWindow *aParent = 0, Window win = 0);
+    YWindow(YWindow *aParent = 0, Window win = 0, Visual *visual = 0);
     virtual ~YWindow();
 
     void setStyle(unsigned long aStyle);
@@ -202,6 +202,8 @@
     bool hasPopup();
     void setDoubleBuffer(bool doubleBuffer);
 
+    Visual *visual() { return fVisual; }
+
 private:
     typedef enum {
         wfVisible   = 1 << 0,
@@ -234,6 +236,7 @@
     YCursor fPointer;
     int unmapCount;
     Graphics *fGraphics;
+    Visual *fVisual;
     long fEventMask;
     int fWinGravity, fBitGravity;
 
