diff -Naur icewm-1.2.28pre2.orig/src/debug.h icewm-1.2.28pre2.patched/src/debug.h
--- icewm-1.2.28pre2.orig/src/debug.h	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/debug.h	2006-09-05 19:17:59.000000000 +0000
@@ -18,4 +18,7 @@
 #define PRECONDITION(x) // nothing
 #endif
 
+#undef MSG
+#define MSG(x) msg x
+
 #endif
diff -Naur icewm-1.2.28pre2.orig/src/misc.cc icewm-1.2.28pre2.patched/src/misc.cc
--- icewm-1.2.28pre2.orig/src/misc.cc	2006-08-31 21:34:43.000000000 +0000
+++ icewm-1.2.28pre2.patched/src/misc.cc	2006-09-05 15:36:44.000000000 +0000
@@ -15,6 +15,20 @@
 #include <libgen.h>
 #endif
 
+#include <sys/time.h>
+#include <time.h>
+void log_time_stamp()
+{
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    struct tm t;
+    localtime_r(&tv.tv_sec, &t);
+    char s[128];
+    memset(&s, 0, sizeof(s));
+    strftime(s, sizeof(s), "%H:%M:%S", &t);
+    fprintf(stderr, "%s.%06d ", s, tv.tv_usec);
+}
+
 extern char const *ApplicationName;
 
 #ifdef DEBUG
@@ -299,6 +313,7 @@
 }
 
 void precondition(char const *msg, ...) {
+    log_time_stamp();
     fprintf(stderr, "%s: ", ApplicationName);
 
     va_list ap;
@@ -311,6 +326,7 @@
 }
 
 void warn(char const *msg, ...) {
+    log_time_stamp();
     fprintf(stderr, "%s: ", ApplicationName);
     fputs(_("Warning: "), stderr);
 
@@ -323,6 +339,7 @@
 }
 
 void msg(char const *msg, ...) {
+    log_time_stamp();
     fprintf(stderr, "%s: ", ApplicationName);
 
     va_list ap;
