Index: libsdl1.2-1.2.9+0/build-scripts/strip_fPIC.sh
===================================================================
--- libsdl1.2-1.2.9+0.orig/build-scripts/strip_fPIC.sh	2001-05-10 22:19:50.000000000 +0200
+++ libsdl1.2-1.2.9+0/build-scripts/strip_fPIC.sh	2006-05-13 18:17:40.000000000 +0200
@@ -5,8 +5,8 @@
 command=""
 while [ $# -gt 0 ]; do
     case "$1" in
-        -?PIC)
-            # Ignore -fPIC and -DPIC options
+        -fPIC)
+            # Ignore -fPIC option
             ;;
         *)
             command="$command $1"
