Description: ?
Author: ?

--- a/config.c
+++ b/config.c
@@ -482,6 +482,8 @@
   ChannelsWrap = 0;
   ShowChannelNamesWithSource = 0;
   EmergencyExit = 1;
+  JumpSeconds = 60;
+  JumpSecondsSlow = 10;
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -696,6 +698,8 @@
   else if (!strcasecmp(Name, "ShowChannelNamesWithSource")) ShowChannelNamesWithSource = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
   else if (!strcasecmp(Name, "LastReplayed"))        cReplayControl::SetRecording(Value);
+  else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSecondsSlow"))     JumpSecondsSlow    = atoi(Value);
   else
      return false;
   return true;
@@ -813,6 +817,8 @@
   Store("ShowChannelNamesWithSource", ShowChannelNamesWithSource);
   Store("EmergencyExit",      EmergencyExit);
   Store("LastReplayed",       cReplayControl::LastReplayed());
+  Store("JumpSeconds",        JumpSeconds);
+  Store("JumpSecondsSlow",    JumpSecondsSlow);
 
   Sort();
 
--- a/config.h
+++ b/config.h
@@ -349,6 +349,8 @@
   int ChannelsWrap;
   int ShowChannelNamesWithSource;
   int EmergencyExit;
+  int JumpSeconds;
+  int JumpSecondsSlow;
   int __EndData__;
   cString InitialChannel;
   cString DeviceBondings;
--- a/menu.c
+++ b/menu.c
@@ -3631,6 +3631,8 @@
   Add(new cMenuEditIntItem( tr("Setup.Replay$Progress display time (s)"), &data.ProgressDisplayTime, 0, 60));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Pause replay when setting mark"), &data.PauseOnMarkSet));
   Add(new cMenuEditIntItem(tr("Setup.Replay$Resume ID"), &data.ResumeID, 0, 99));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds"), &data.JumpSeconds));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds Slow"), &data.JumpSecondsSlow));
 }
 
 void cMenuSetupReplay::Store(void)
@@ -5590,9 +5592,9 @@
     case kRight:   Forward(); break;
     case kRed:     TimeSearch(); break;
     case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-60); break;
+    case kGreen:   SkipSeconds(-Setup.JumpSeconds); break;
     case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 60); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
     case kStop:
     case kBlue:    Hide();
                    Stop();
