=== modified file 'debian/changelog'
--- debian/changelog	2008-06-12 22:51:27 +0000
+++ debian/changelog	2008-07-22 19:03:48 +0000
@@ -1,3 +1,9 @@
+mythplugins (0.22.0~trunk17891-0ubuntu1) intrepid; urgency=low
+
+  * update to 17891
+
+ -- Daniel Kristjansson <danielk@cuymedia.net> Tue, 22 Jul 2008 15:02:00 -0500
+
 mythplugins (0.22.0~trunk17463-0ubuntu1) hardy; urgency=low
 
   * first trunk build (0.22 series) 

=== modified file 'debian/control'
--- debian/control	2008-06-12 22:51:27 +0000
+++ debian/control	2008-07-22 19:01:08 +0000
@@ -6,7 +6,7 @@
 XSBC-Original-Maintainer: Christian Marillat <marillat@debian.org>
 Standards-Version: 3.7.3
 Build-Depends: debhelper (>> 5), 
-               libmyth-dev (>= 0.22.0~trunk17463-0ubuntu1), 
+               libmyth-dev (>= 0.22.0~trunk17891-0ubuntu1), 
                dpatch, 
                libgl1-mesa-dev | xlibmesa-gl-dev | xlibmesa-dev | libgl-dev, 
                libdvdread3-dev, 

=== modified file 'debian/patches/00list'
--- debian/patches/00list	2008-06-12 22:51:27 +0000
+++ debian/patches/00list	2008-07-22 00:35:48 +0000
@@ -7,4 +7,4 @@
 09_mythvideo_ubuntu_directories.dpatch
 10_mythmusic_faad
 06_mytharchive_ubuntu_directories
-12_missing_malloc.dpatch
+#12_missing_malloc.dpatch

=== modified file 'debian/rules'
--- debian/rules	2008-06-12 22:51:27 +0000
+++ debian/rules	2008-07-22 19:27:56 +0000
@@ -7,7 +7,7 @@
 SVN_TYPE+=trunk
 SVN_MAJOR_RELEASE+=22
 SVN_MINOR_RELEASE+=0
-SVN_REVISION+=17463
+SVN_REVISION+=17891
 #SVN_BRANCH+= http://svn.mythtv.org/svn/branches/release-0-$(SVN_MAJOR_RELEASE)-$(SVN_TYPE)/$(SVN_PACKAGE)
 SVN_BRANCH+= http://svn.mythtv.org/svn/$(SVN_TYPE)/$(SVN_PACKAGE)
 GREATER+=false
@@ -32,6 +32,9 @@
 	tar czf $(CURDIR)/../$(TARFILE) $(SVN_PACKAGE)-$(SVN_RELEASE)$(SUFFIX)
 	rm -rf $(CURDIR)/$(SVN_PACKAGE)-$(SVN_RELEASE)$(SUFFIX)
 
+extract-source:
+	tar xvf mythplugins-$(SVN_TYPE)-r$(SVN_REVISION).tbz
+
 NCPUS := $(shell getconf _NPROCESSORS_ONLN 2>/dev/null || echo 1)
 
 ifeq ($(NCPUS),-1)
@@ -52,6 +55,8 @@
 
 export QTDIR=$(shell for qtdir in /usr/share/qt3 /usr/share/qt; do if test -d $$qtdir; then echo $$qtdir; break; fi; done)
 
+patch-stamp: extract-source patch
+
 configure: configure-stamp
 configure-stamp: patch-stamp
 
@@ -60,9 +65,9 @@
 
 	dh_testdir
 
-	./configure --prefix=/usr --enable-all --disable-festival --zm-version=1.22.3
+	./configure --prefix=/usr --enable-all --disable-festival --zm-version=1.22.3 --qmake=qmake-qt4
 
-	qmake -o Makefile PREFIX=/usr mythplugins.pro
+	qmake-qt4 -o Makefile PREFIX=/usr mythplugins.pro
 
 	touch configure-stamp
 
@@ -85,7 +90,7 @@
 	rm -f config.mak config.h
 	rm -f debian/shlibs.local
 
-	debian/rules unpatch
+#	debian/rules unpatch
 
 	dh_clean Makefile config.pro mythgallery/mythgallery/config.h \
 	mythmusic/mythmusic/config.h mythgallery/mythgallery/config.pro \
@@ -95,6 +100,14 @@
 	config.log mythconfig.mak mytharchive/mytharchive/config.h \
 	mytharchive/mytharchive/config.pro
 
+	rm -Rf COPYING Makefile README cleanup config.log \
+	       config.pro configure configure-stamp libs-targetfix.pro \
+	       mytharchive mythconfig.mak mythcontrols mythflix \
+	       mythgallery mythgame mythmovies mythmusic mythnews \
+	       mythphone mythplugins.pro mythvideo mythweather \
+	       mythweb mythzoneminder patch-stamp programs-libs.pro \
+	       settings.pro themedummy.c
+
 install: build
 	dh_testdir
 	dh_testroot

