--- a/depmod.c	2005-11-21 04:45:15.000000000 +0200
+++ b/depmod.c	2006-09-11 01:28:40.000000000 +0300
@@ -477,11 +477,19 @@
 static int is_update(const char *dirname)
 {
 	char *p;
+	size_t len;
 
 	p = strstr(dirname, "updates");
+	len = strlen("updated");
+	if (!p)
+	{
+		p = strstr(dirname, "misc");
+		len = strlen("misc");
+	}
 	if (!p)
 		return 0;
-	return (p[strlen("updates")] == '/' || p[strlen("updates")] == '\0');
+
+	return (p[len] == '/' || p[len] == '\0');
 }
 
 /* Grab everything not under updates/ directories. */
@@ -576,6 +584,14 @@
 	list = grab_dir(dirname, dir, NULL, do_normal_module);
 	closedir(dir);
 
+	/* Pluto-ism: misc/ dir overrides other modules */
+	sprintf(updatedir, "%s/misc", dirname);
+	dir = opendir(updatedir);
+	if (dir) {
+		list = grab_dir(updatedir, dir, list, do_update_module);
+		closedir(dir);
+	}
+	
 	sprintf(updatedir, "%s/updates", dirname);
 	dir = opendir(updatedir);
 	if (dir) {
